From 8bfdc5150cabb1e7a5c4f970ccf0f4374ac8841d Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 15:56:30 +0100
Subject: [PATCH 1/5] build: Fix logic for --with-x configure option

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index fa75e930..847e95fb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1115,7 +1115,7 @@ if test "x$dbus_win" = xyes; then
 
     enable_x11_autolaunch=no
     have_x11=no
-else if test "x$with_x" = xauto; then
+else if test "x$with_x" != xno; then
     PKG_CHECK_MODULES([X], [x11],
             [AC_DEFINE([HAVE_X11], [1], [Define to 1 if you have X11 library])],
             [ have_x11=no ])
-- 
2.35.1


From ea4692b61b2992609f3ae4a52624c729955105c1 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 16:01:29 +0100
Subject: [PATCH 2/5] Move DBUS_X_* definitions into PKG_CHECK_MODULES block

---
 configure.ac | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 847e95fb..a51e7265 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1117,15 +1117,15 @@ if test "x$dbus_win" = xyes; then
     have_x11=no
 else if test "x$with_x" != xno; then
     PKG_CHECK_MODULES([X], [x11],
-            [AC_DEFINE([HAVE_X11], [1], [Define to 1 if you have X11 library])],
+            [AC_DEFINE([HAVE_X11], [1], [Define to 1 if you have X11 library])
+	     have_x11=yes
+	     DBUS_X_LIBS="$X_LIBS"
+	     DBUS_X_CFLAGS="$X_CFLAGS"
+            ],
             [ have_x11=no ])
 
     if test "x$have_x11" = xno; then
         AC_MSG_WARN([Couldn't found X11, tried with pkg-config.])
-    else
-        have_x11=yes
-        DBUS_X_LIBS="$X_LIBS"
-        DBUS_X_CFLAGS="$X_CFLAGS"
     fi
 else
     AS_IF([test "x$enable_x11_autolaunch" = "xyes"], [
-- 
2.35.1


From b8a57d110a2f32f68c746fc0912f179194d17003 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 16:05:52 +0100
Subject: [PATCH 3/5] Emit an error when --with-x was given but no X11 libs
 were found

---
 configure.ac | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index a51e7265..4420c6d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1121,12 +1121,13 @@ else if test "x$with_x" != xno; then
 	     have_x11=yes
 	     DBUS_X_LIBS="$X_LIBS"
 	     DBUS_X_CFLAGS="$X_CFLAGS"
-            ],
-            [ have_x11=no ])
-
-    if test "x$have_x11" = xno; then
-        AC_MSG_WARN([Couldn't found X11, tried with pkg-config.])
-    fi
+            ], [
+             AS_IF([test "x$with_x" = xyes],
+                [AC_MSG_ERROR([Couldn't find X11, tried with pkg-config.])],
+                [AC_MSG_WARN([Couldn't find X11, tried with pkg-config.])]
+             )
+             have_x11=no
+            ])
 else
     AS_IF([test "x$enable_x11_autolaunch" = "xyes"], [
       AC_MSG_ERROR([--enable-x11-autolaunch and --without-x are not compatible])
-- 
2.35.1


From 58d16b49ff78d1fb5ec92eae23fc1b0bc8dfb5fd Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 16:08:06 +0100
Subject: [PATCH 4/5] Add have_x11=no if --without-x was given to make later
 checks happy

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 4420c6d3..8816a950 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1129,6 +1129,7 @@ else if test "x$with_x" != xno; then
              have_x11=no
             ])
 else
+    have_x11=no
     AS_IF([test "x$enable_x11_autolaunch" = "xyes"], [
       AC_MSG_ERROR([--enable-x11-autolaunch and --without-x are not compatible])
     ])
-- 
2.35.1


From f425b564be30b4fd68e772ea94dbeacc4d0ff3a2 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 16:11:05 +0100
Subject: [PATCH 5/5] Convert "if" calls to AS_IF macro for the block that
 handles X11

---
 configure.ac | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8816a950..91bd898c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1108,14 +1108,14 @@ AC_ARG_ENABLE([x11-autolaunch],
   AS_HELP_STRING([--enable-x11-autolaunch], [build with X11 auto-launch support]),
   [], [enable_x11_autolaunch=auto])
 
-if test "x$dbus_win" = xyes; then
-    if test "x$enable_x11_autolaunch" = xyes; then
+AS_IF([test "x$dbus_win" = xyes], [
+    AS_IF([test "x$enable_x11_autolaunch" = xyes], [
         AC_MSG_ERROR([X11 auto-launch is not supported on Windows])
-    fi
+    ])
 
     enable_x11_autolaunch=no
     have_x11=no
-else if test "x$with_x" != xno; then
+], [test "x$with_x" != xno], [
     PKG_CHECK_MODULES([X], [x11],
             [AC_DEFINE([HAVE_X11], [1], [Define to 1 if you have X11 library])
 	     have_x11=yes
@@ -1128,13 +1128,12 @@ else if test "x$with_x" != xno; then
              )
              have_x11=no
             ])
-else
+], [
     have_x11=no
     AS_IF([test "x$enable_x11_autolaunch" = "xyes"], [
       AC_MSG_ERROR([--enable-x11-autolaunch and --without-x are not compatible])
     ])
-fi
-fi
+])
 
 if test "x$enable_x11_autolaunch,$have_x11" = xyes,no; then
     AC_MSG_ERROR([X11 auto-launch requires X headers/libraries])
-- 
2.35.1

