--- a/cmake/external/glib/generate_cppgir.cmake	2024-01-21 10:25:18.445073339 +0700
+++ b/cmake/external/glib/generate_cppgir.cmake	2024-01-21 10:28:55.563656610 +0700
@@ -4,6 +4,8 @@
 # For license and copyright information please follow this link:
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
+find_package(CppGir REQUIRED)
+
 function(generate_cppgir target_name gir)
     set(cppgir_loc ${cmake_helpers_loc}/external/glib/cppgir)
 
