From 1d1f6a5fa57bdbe1889c0f2a17533fbf0c512531 Mon Sep 17 00:00:00 2001
From: John Preston <johnprestonmail@gmail.com>
Date: Thu, 13 Jan 2022 16:01:12 +0300
Subject: [PATCH] Allow external libvpx.

---
 cmake/external.cmake | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/cmake/external.cmake b/cmake/external.cmake
index c11df5b6..1f14b359 100644
--- a/cmake/external.cmake
+++ b/cmake/external.cmake
@@ -161,7 +161,12 @@ function(link_libusrsctp target_name)
 endfunction()
 
 # libvpx
+set(TG_OWT_LIBVPX_INSTALL_PATH "" CACHE STRING "Install path for libvpx.")
 function(link_libvpx target_name)
-    if (TG_OWT_PACKAGED_BUILD)
+    if (TG_OWT_PACKAGED_BUILD
+        OR NOT TG_OWT_LIBVPX_INSTALL_PATH STREQUAL "")
         find_package(PkgConfig REQUIRED)
+        if (NOT TG_OWT_LIBVPX_INSTALL_PATH STREQUAL "")
+            set(CMAKE_PREFIX_PATH "${TG_OWT_LIBVPX_INSTALL_PATH}/lib/pkgconfig")
+        endif()
         pkg_check_modules(LIBVPX vpx>=1.10.0)
