diff -up tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc.omv~ tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc
--- tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc.omv~	2025-02-26 14:13:53.388804545 +0100
+++ tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc	2025-02-26 14:14:58.178031834 +0100
@@ -60,7 +60,7 @@ PipeWireNode::PipeWireNode(PipeWireSessi
       .param = OnNodeParam,
   };
 
-  pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
+  pw_node_add_listener(reinterpret_cast<pw_node*>(proxy_), &node_listener_, &node_events, this);
 }
 
 PipeWireNode::~PipeWireNode() {
@@ -94,7 +94,7 @@ void PipeWireNode::OnNodeInfo(void* data
       uint32_t id = info->params[i].id;
       if (id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
+        pw_node_enum_params(reinterpret_cast<pw_node*>(that->proxy_), 0, id, 0, UINT32_MAX, nullptr);
         break;
       }
     }
