diff -up libreoffice-7.6.0.1/vcl/qt5/QtFrame.cxx.omv~ libreoffice-7.6.0.1/vcl/qt5/QtFrame.cxx
--- libreoffice-7.6.0.1/vcl/qt5/QtFrame.cxx.omv~	2023-07-28 12:12:58.638604442 +0200
+++ libreoffice-7.6.0.1/vcl/qt5/QtFrame.cxx	2023-07-28 12:13:49.152094195 +0200
@@ -202,7 +202,7 @@ void QtFrame::FillSystemEnvData(SystemEn
 {
     assert(rData.platform == SystemEnvData::Platform::Invalid);
     assert(rData.toolkit == SystemEnvData::Toolkit::Invalid);
-    if (QGuiApplication::platformName() == "wayland")
+    if (QGuiApplication::platformName().startsWith("wayland")) // wayland or wayland-egl
         rData.platform = SystemEnvData::Platform::Wayland;
     else if (QGuiApplication::platformName() == "xcb")
         rData.platform = SystemEnvData::Platform::Xcb;
