--- LVM2.2.03.00/libdm/libdevmapper.pc.in
+++ LVM2.2.03.00/libdm/libdevmapper.pc.in
@@ -9,4 +9,4 @@
 Cflags: -I${includedir} 
 Libs: -L${libdir} -ldevmapper
 Requires.private: @SELINUX_PC@ @UDEV_PC@
-Libs.private: -lm @RT_LIBS@
+Libs.private: -lm @RT_LIBS@ @PTHREAD_LIBS@
--- LVM2.2.03.00/tools/Makefile.in
+++ LVM2.2.03.00/tools/Makefile.in
@@ -87,6 +87,7 @@
   TARGETS += lvm.static
   INSTALL_LVM_TARGETS += install_tools_static
   INSTALL_CMDLIB_TARGETS += install_cmdlib_static
+  STATIC_LIBS += @PTHREAD_LIBS@
 endif
 
 LVMLIBS = $(LVMINTERNAL_LIBS) -laio
@@ -112,6 +113,10 @@
 
 include $(top_builddir)/make.tmpl
 
+ifeq ("@STATIC_LINK@", "yes")
+  STATIC_LIBS += @PTHREAD_LIBS@
+endif
+
 device-mapper:
 
 all: device-mapper
