diff -Nuar a/Makefile.am b/Makefile.am
--- a/Makefile.am	2016-06-19 18:26:18.000000000 +0700
+++ b/Makefile.am	2017-10-11 17:17:41.981581065 +0700
@@ -59,9 +59,7 @@
 docsdir = $(docdir)
 
 docs_DATA = README help.txt about.txt translators.txt ChangeLog\
-	COPYING.GPLv2 COPYING.LGPLv2.1\
-	translation/help.ru.txt\
-	translation/help.pt_BR.txt
+	COPYING.GPLv2 COPYING.LGPLv2.1
 
 desktopdir = $(datadir)/applications
 desktop_DATA = deadbeef.desktop
diff -Nuar a/Makefile.in b/Makefile.in
--- a/Makefile.in	2016-06-19 18:30:46.000000000 +0700
+++ b/Makefile.in	2017-10-11 17:17:50.661680273 +0700
@@ -515,9 +515,7 @@
 AM_CPPFLAGS = $(DEPS_CFLAGS)
 docsdir = $(docdir)
 docs_DATA = README help.txt about.txt translators.txt ChangeLog\
-	COPYING.GPLv2 COPYING.LGPLv2.1\
-	translation/help.ru.txt\
-	translation/help.pt_BR.txt
+	COPYING.GPLv2 COPYING.LGPLv2.1
 
 desktopdir = $(datadir)/applications
 desktop_DATA = deadbeef.desktop
