--- /usr/share/plasma/plasmoids/org.kde.plasma.volume/contents/ui/ListItemBase.qml	2016-01-27 02:43:19.000000000 +0700
+++ /usr/share/plasma/plasmoids/org.kde.plasma.volume/contents/ui/ListItemBase.qml	2016-02-09 21:03:33.607016292 +0700
@@ -59,14 +59,14 @@
     //        from this, they are not used there though.
     //        seems naughty.
     function increaseVolume() {
-        var step = slider.maximumValue / 15;
+        var step = slider.maximumValue / 115;
         var volume = bound(PulseObject.volume + step, 0, slider.maximumValue);
         setVolume(volume);
         osd.show(volumePercent(volume));
     }
 
     function decreaseVolume() {
-        var step = slider.maximumValue / 15;
+        var step = slider.maximumValue / 115;
         var volume = bound(PulseObject.volume - step, 0, slider.maximumValue);
         setVolume(volume);
         osd.show(volumePercent(volume));
