--- nasm-2.14/Makefile.in
+++ nasm-2.14/Makefile.in
@@ -74,13 +74,13 @@
 .PHONY: manpages nsis
 
 .c.$(O):
-	$(CC) -c $(ALL_CFLAGS) -o $@ $<
+	$(CC) -c $(CPPFLAGS) $(ALL_CFLAGS) -o $@ $<
 
 .c.s:
-	$(CC) -S $(ALL_CFLAGS) -o $@ $<
+	$(CC) -S $(CPPFLAGS) $(ALL_CFLAGS) -o $@ $<
 
 .c.i:
-	$(CC) -E $(ALL_CFLAGS) -o $@ $<
+	$(CC) -E $(CPPFLAGS) $(ALL_CFLAGS) -o $@ $<
 
 .txt.xml:
 	$(ASCIIDOC) -b docbook -d manpage -o $@ $<
