diff -up firefox-76.0/old-configure.in.nss-version firefox-76.0/old-configure.in
--- firefox-76.0/old-configure.in.nss-version	2020-05-03 19:12:01.771863688 +0200
+++ firefox-76.0/old-configure.in	2020-05-03 19:17:53.216898766 +0200
@@ -1506,7 +1506,7 @@ MOZ_ARG_WITH_BOOL(system-nss,
     _USE_SYSTEM_NSS=1 )
 
 if test -n "$_USE_SYSTEM_NSS"; then
-    AM_PATH_NSS(3.51.1, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
+    AM_PATH_NSS(3.51, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
 NSS_CFLAGS="$NSS_CFLAGS -I${DIST}/include/nss"
