diff -rupN sqlitebrowser-3.12.0/CMakeLists.txt sqlitebrowser-3.12.0-new/CMakeLists.txt
--- sqlitebrowser-3.12.0/CMakeLists.txt	2020-06-14 06:08:29.000000000 +0200
+++ sqlitebrowser-3.12.0-new/CMakeLists.txt	2020-06-16 10:29:34.518119985 +0200
@@ -461,7 +461,7 @@ endif()
 if(QSCINTILLA_FOUND)
     target_link_libraries(${PROJECT_NAME} ${QSCINTILLA_LIBRARIES})
 else()
-    target_link_libraries(${PROJECT_NAME} qscintilla2)
+    target_link_libraries(${PROJECT_NAME} qscintilla2_qt5)
 endif()
 
 if(WIN32 AND MSVC)
