diff -U0 firefox-85.0.1/browser/components/enterprisepolicies/tests/xpcshell/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/browser/components/enterprisepolicies/tests/xpcshell/xpcshell.ini
--- firefox-85.0.1/browser/components/enterprisepolicies/tests/xpcshell/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:46.000000000 +0100
+++ firefox-85.0.1/browser/components/enterprisepolicies/tests/xpcshell/xpcshell.ini	2021-02-23 17:26:03.907321729 +0100
@@ -9 +8,0 @@
-[test_appupdateurl.js]
diff -U0 firefox-85.0.1/devtools/client/shared/remote-debugging/adb/xpcshell/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/devtools/client/shared/remote-debugging/adb/xpcshell/xpcshell.ini
--- firefox-85.0.1/devtools/client/shared/remote-debugging/adb/xpcshell/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:47.000000000 +0100
+++ firefox-85.0.1/devtools/client/shared/remote-debugging/adb/xpcshell/xpcshell.ini	2021-02-23 17:26:03.908321759 +0100
@@ -9 +8,0 @@
-[test_adb.js]
diff -U0 firefox-85.0.1/netwerk/test/unit_ipc/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/netwerk/test/unit_ipc/xpcshell.ini
--- firefox-85.0.1/netwerk/test/unit_ipc/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:52.000000000 +0100
+++ firefox-85.0.1/netwerk/test/unit_ipc/xpcshell.ini	2021-02-23 17:26:03.908321759 +0100
@@ -73 +72,0 @@
-[test_dns_service_wrap.js]
diff -U0 firefox-85.0.1/netwerk/test/unit/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/netwerk/test/unit/xpcshell.ini
--- firefox-85.0.1/netwerk/test/unit/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 22:06:36.000000000 +0100
+++ firefox-85.0.1/netwerk/test/unit/xpcshell.ini	2021-02-23 17:26:03.908321759 +0100
@@ -202 +201,0 @@
-[test_dns_service.js]
@@ -226 +224,0 @@
-[test_file_protocol.js]
@@ -334 +331,0 @@
-[test_unix_domain.js]
@@ -346 +342,0 @@
-[test_udp_multicast.js]
@@ -395,2 +390,0 @@
-[test_tls_flags.js]
-skip-if = (verify && (os == 'linux')) || (os == "android" && processor == "x86_64")
@@ -412 +405,0 @@
-[test_network_connectivity_service.js]
diff -U0 firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell.ini
--- firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 22:06:36.000000000 +0100
+++ firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell.ini	2021-02-23 17:26:03.908321759 +0100
@@ -183 +182,0 @@
-[test_oskeystore.js]
diff -U0 firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell-smartcards.ini.firefox-tests-xpcshell firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell-smartcards.ini
--- firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell-smartcards.ini.firefox-tests-xpcshell	2021-02-04 16:52:52.000000000 +0100
+++ firefox-85.0.1/security/manager/ssl/tests/unit/xpcshell-smartcards.ini	2021-02-23 17:26:03.908321759 +0100
@@ -10 +9,0 @@
-[test_pkcs11_module.js]
diff -U0 firefox-85.0.1/toolkit/components/commandlines/test/unit/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/toolkit/components/commandlines/test/unit/xpcshell.ini
--- firefox-85.0.1/toolkit/components/commandlines/test/unit/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:57.000000000 +0100
+++ firefox-85.0.1/toolkit/components/commandlines/test/unit/xpcshell.ini	2021-02-23 17:26:03.908321759 +0100
@@ -10 +9,0 @@
-[test_resolvefile.js]
diff -U0 firefox-85.0.1/toolkit/components/extensions/test/xpcshell/xpcshell-common.ini.firefox-tests-xpcshell firefox-85.0.1/toolkit/components/extensions/test/xpcshell/xpcshell-common.ini
--- firefox-85.0.1/toolkit/components/extensions/test/xpcshell/xpcshell-common.ini.firefox-tests-xpcshell	2021-02-04 22:06:37.000000000 +0100
+++ firefox-85.0.1/toolkit/components/extensions/test/xpcshell/xpcshell-common.ini	2021-02-23 17:26:03.908321759 +0100
@@ -76,2 +75,0 @@
-[test_ext_downloads_misc.js]
-skip-if = os == "android" || (os=='linux' && bits==32) || tsan # linux32: bug 1324870, tsan: bug 1612707
@@ -244,2 +241,0 @@
-[test_proxy_listener.js]
-skip-if = appname == "thunderbird"
diff -U0 firefox-85.0.1/toolkit/modules/subprocess/test/xpcshell/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/toolkit/modules/subprocess/test/xpcshell/xpcshell.ini
--- firefox-85.0.1/toolkit/modules/subprocess/test/xpcshell/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:57.000000000 +0100
+++ firefox-85.0.1/toolkit/modules/subprocess/test/xpcshell/xpcshell.ini	2021-02-23 17:26:03.909321789 +0100
@@ -10 +9,0 @@
-[test_subprocess.js]
diff -U0 firefox-85.0.1/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini.firefox-tests-xpcshell firefox-85.0.1/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini
--- firefox-85.0.1/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini.firefox-tests-xpcshell	2021-02-04 16:52:57.000000000 +0100
+++ firefox-85.0.1/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini	2021-02-23 17:26:03.909321789 +0100
@@ -97 +96,0 @@
-[test_sideloads_after_rebuild.js]
