--- a/einstein-2.0/Makefile	2005-09-25 23:20:30.000000000 +0200
+++ b/einstein-2.0/Makefile	2011-09-25 13:35:27.000000000 +0200
@@ -5,7 +5,7 @@
 ########################################
 
 # installation prefix
-PREFIX=/usr/local
+PREFIX=/usr
 
 ########################################
 #
@@ -49,10 +49,14 @@
 
 
 $(TARGET): $(OBJECTS)
+	cd mkres && make
+	cd res && ../mkres/mkres --source resources.descr --output einstein.res
 	$(CXX) $(LNFLAGS) $(OBJECTS) -o $(TARGET)
 
 clean:
 	rm -f $(OBJECTS) core* *core $(TARGET) *~
+	cd res && rm -f einstein.res
+	cd mkres && make clean
 
 depend:
 	@makedepend $(SOURCES) 2> /dev/null
@@ -61,8 +65,7 @@
 	./$(TARGET)
 
 install: $(TARGET)
-	$(INSTALL) -s -D $(TARGET) $(PREFIX)/bin/$(TARGET)
-	$(INSTALL) -D einstein.res $(PREFIX)/share/einstein/res/einstein.res
+	$(INSTALL) -s -D $(TARGET) $(DESTDIR)/$(PREFIX)/games/$(TARGET)
 	
 # DO NOT DELETE THIS LINE -- make depend depends on it.
 
