--- a/polly/cmake/CMakeLists.txt	2022-09-14 17:43:15.838596458 +0700
+++ b/polly/cmake/CMakeLists.txt	2022-09-14 17:44:59.008837644 +0700
@@ -1,9 +1,5 @@
 # Keep this in sync with llvm/cmake/CMakeLists.txt!
 
-include(GNUInstallPackageDir)
-include(ExtendPath)
-include(FindPrefixFromConfig)
-
 set(POLLY_INSTALL_PACKAGE_DIR "${CMAKE_INSTALL_PACKAGEDIR}/polly" CACHE STRING
   "Path for CMake subdirectory for Polly (defaults to '${CMAKE_INSTALL_PACKAGEDIR}/polly')")
 # CMAKE_INSTALL_PACKAGEDIR might be absolute, so don't reuse below.
@@ -90,11 +86,6 @@
 
 # Generate PollyConfig.cmake for the install tree.
 unset(POLLY_EXPORTS)
-find_prefix_from_config(POLLY_CONFIG_CODE POLLY_INSTALL_PREFIX "${POLLY_INSTALL_PACKAGE_DIR}")
-extend_path(POLLY_CONFIG_LLVM_CMAKE_DIR "\${POLLY_INSTALL_PREFIX}" "${LLVM_INSTALL_PACKAGE_DIR}")
-extend_path(POLLY_CONFIG_CMAKE_DIR "\${POLLY_INSTALL_PREFIX}" "${POLLY_INSTALL_PACKAGE_DIR}")
-extend_path(POLLY_CONFIG_LIBRARY_DIRS "\${POLLY_INSTALL_PREFIX}" "lib${LLVM_LIBDIR_SUFFIX}")
-extend_path(base_includedir "\${POLLY_INSTALL_PREFIX}" "${CMAKE_INSTALL_INCLUDEDIR}")
 if (POLLY_BUNDLED_ISL)
   set(POLLY_CONFIG_INCLUDE_DIRS
     "${base_includedir}"
@@ -124,8 +115,6 @@
     set(tgt_prefix "${CMAKE_INSTALL_LIBDIR}")
   endif()
 
-  extend_path(tgt_path "\${POLLY_INSTALL_PREFIX}" "${tgt_prefix}/$<TARGET_FILE_NAME:${tgt}>")
-
   if (NOT tgt_type STREQUAL "INTERFACE_LIBRARY")
     set(POLLY_EXPORTS
 "set_target_properties(${tgt} PROPERTIES
