--- ./chrome/browser/ui/toolbar/wrench_menu_model.cc
+++ ./chrome/browser/ui/toolbar/wrench_menu_model.cc.new
@@ -930,21 +930,7 @@
   CreateCutCopyPasteMenu();
 
   AddItemWithStringId(IDC_OPTIONS, IDS_SETTINGS);
-#if !defined(OS_CHROMEOS)
-  if (!switches::IsNewAvatarMenu()) {
-    // No "Sign in to Chromium..." menu item on ChromeOS.
-    SigninManager* signin = SigninManagerFactory::GetForProfile(
-        browser_->profile()->GetOriginalProfile());
-    if (signin && signin->IsSigninAllowed() &&
-        signin_ui_util::GetSignedInServiceErrors(
-            browser_->profile()->GetOriginalProfile()).empty()) {
-      AddItem(IDC_SHOW_SYNC_SETUP,
-              l10n_util::GetStringFUTF16(
-                  IDS_SYNC_MENU_PRE_SYNCED_LABEL,
-                  l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME)));
-    }
-  }
-#endif
+
 // The help submenu is only displayed on official Chrome builds. As the
 // 'About' item has been moved to this submenu, it's reinstated here for
 // Chromium builds.

