Index: chromium-69.0.3472.3/sandbox/linux/BUILD.gn
===================================================================
--- chromium-69.0.3472.3.orig/sandbox/linux/BUILD.gn
+++ chromium-69.0.3472.3/sandbox/linux/BUILD.gn
@@ -314,12 +314,15 @@ if (is_linux) {
     cflags = [
       # For ULLONG_MAX
       "-std=gnu99",
+      "-fPIE",
 
       # These files have a suspicious comparison.
       # TODO fix this and re-enable this warning.
       "-Wno-sign-compare",
     ]
 
+    ldflags = [ "-pie" ]
+
     import("//build/config/compiler/compiler.gni")
     import("//build/config/sanitizers/sanitizers.gni")
     if (is_component_build || using_sanitizer) {
@@ -329,7 +332,7 @@ if (is_linux) {
       # other flags that executable_config might have.
       configs -= [ "//build/config:executable_config" ]
       if (!use_gold) {
-        ldflags = [ "-Wl,--disable-new-dtags" ]
+        ldflags += [ "-Wl,--disable-new-dtags" ]
       }
     }
 
