diff -Nuar llvm/cmake/modules/AddLLVM.cmake llvm_/cmake/modules/AddLLVM.cmake
--- llvm/cmake/modules/AddLLVM.cmake	2020-07-28 18:47:17.000000000 +0700
+++ llvm_/cmake/modules/AddLLVM.cmake	2020-08-03 20:57:19.614460292 +0700
@@ -963,10 +963,10 @@
   endif()
 
   # Add static plugins to the Extension component
-  foreach(llvm_extension ${LLVM_STATIC_EXTENSIONS})
-      set_property(TARGET LLVMExtensions APPEND PROPERTY LINK_LIBRARIES ${llvm_extension})
-      set_property(TARGET LLVMExtensions APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${llvm_extension})
-  endforeach()
+  #foreach(llvm_extension ${LLVM_STATIC_EXTENSIONS})
+  #    set_property(TARGET LLVMExtensions APPEND PROPERTY LINK_LIBRARIES ${llvm_extension})
+  #    set_property(TARGET LLVMExtensions APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${llvm_extension})
+  #endforeach()
 
   # Eventually generate the extension headers, and store config to a cmake file
   # for usage in third-party configuration.
