From fbb3f41dff9ad6499de48cadd756e99f18b4392c Mon Sep 17 00:00:00 2001
From: gabime <gmelman1@gmail.com>
Date: Sun, 20 Oct 2019 10:22:44 +0300
Subject: [PATCH] Fix #1273

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa9e1fab..abceb307 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,11 +82,15 @@ message(STATUS "Build type: " ${CMAKE_BUILD_TYPE})
 set(SPDLOG_SRCS
         src/spdlog.cpp
         src/stdout_sinks.cpp
-        src/fmt.cpp
         src/color_sinks.cpp
         src/file_sinks.cpp
         src/async.cpp)
 
+
+if(NOT SPDLOG_FMT_EXTERNAL)
+    list(APPEND SPDLOG_SRCS src/fmt.cpp)
+endif()
+
 if (SPDLOG_BUILD_SHARED)
     if(WIN32)
         message(FATAL_ERROR "spdlog shared lib is not yet supported under windows")
