diff -up firefox-75.0/old-configure.in.nss-version firefox-75.0/old-configure.in
--- firefox-75.0/old-configure.in.nss-version	2020-04-07 07:59:28.091106813 +0200
+++ firefox-75.0/old-configure.in	2020-04-07 08:00:30.272678298 +0200
@@ -1504,7 +1504,7 @@ MOZ_ARG_WITH_BOOL(system-nss,
     _USE_SYSTEM_NSS=1 )
 
 if test -n "$_USE_SYSTEM_NSS"; then
-    AM_PATH_NSS(3.51, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
+    AM_PATH_NSS(3.50, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
 NSS_CFLAGS="$NSS_CFLAGS -I${DIST}/include/nss"
