diff -up firefox-65.0/layout/base/PresShell.h.1354671 firefox-65.0/layout/base/PresShell.h
--- firefox-65.0/layout/base/PresShell.h.1354671	2019-01-29 11:50:08.516649643 +0100
+++ firefox-65.0/layout/base/PresShell.h	2019-01-29 11:51:48.930258629 +0100
@@ -205,7 +205,7 @@ class nsIPresShell : public nsStubDocume
    * to the same aSize value.  AllocateFrame is infallible and will abort
    * on out-of-memory.
    */
-  void* AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize) {
+  void* __attribute__((optimize("no-lifetime-dse"))) AllocateFrame(nsQueryFrame::FrameIID aID, size_t aSize) {
     void* result = mFrameArena.AllocateByFrameID(aID, aSize);
     RecordAlloc(result);
     return result;
