--- a/Makefile	2020-01-11 17:18:03.772531703 +0700
+++ b/Makefile	2020-01-11 17:18:53.061153146 +0700
@@ -67,7 +67,7 @@
 ifndef NO_VENV
 	@echo "Creating our virtualenv"
 	${PYTHON} -m venv env
-	$(VENV_PYTHON) -m pip install -r requirements.txt
+	$(VENV_PYTHON) -m pip install --user -r requirements.txt
 # We can't use the "--system-site-packages" flag on creation because otherwise we end up with
 # the system's pip and that messes up things in some cases (notably in Gentoo).
 	${PYTHON} -m venv --upgrade ${VENV_OPTIONS} env
