diff -urN freetype-2.6.2.orig/src/autofit/afmodule.c freetype-2.6.2/src/autofit/afmodule.c
--- freetype-2.6.2.orig/src/autofit/afmodule.c	2015-12-07 12:42:05.976290812 +0000
+++ freetype-2.6.2/src/autofit/afmodule.c	2015-12-08 09:56:13.342393047 +0000
@@ -22,6 +22,10 @@
 #include "aferrors.h"
 #include "afpic.h"
 
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+#include "../base/ftinf.h"
+#endif
+
 #ifdef FT_DEBUG_AUTOFIT
 
 #ifndef FT_MAKE_OPTION_SINGLE_OBJECT
@@ -372,6 +376,9 @@
     module->warping           = 0;
 #endif
     module->no_stem_darkening = TRUE;
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+    if(ftinf) module->no_stem_darkening = !ftinf->stem_darkening_autofit;
+#endif /* FT_CONFIG_OPTION_INFINALITY_PATCHSET */
 
     module->darken_params[0]  = CFF_CONFIG_OPTION_DARKENING_PARAMETER_X1;
     module->darken_params[1]  = CFF_CONFIG_OPTION_DARKENING_PARAMETER_Y1;
diff -urN freetype-2.6.2.orig/src/base/ftinf.c freetype-2.6.2/src/base/ftinf.c
--- freetype-2.6.2.orig/src/base/ftinf.c	2015-12-07 12:42:05.976290812 +0000
+++ freetype-2.6.2/src/base/ftinf.c	2015-12-08 09:55:29.149551250 +0000
@@ -235,6 +235,10 @@
     if( s ) sscanf( s, "%d", &_env.stem_snapping_sliding_scale );
     s=getenv( "INFINALITY_FT_STEM_ALIGNMENT_STRENGTH" );
     if( s ) sscanf( s, "%d", &_env.stem_alignment_strength );
+    s=getenv( "INFINALITY_FT_STEM_DARKENING_AUTOFIT" );
+    if( s ) _env.stem_darkening_autofit=bool_val( s );
+    s=getenv( "INFINALITY_FT_STEM_DARKENING_CFF" );
+    if( s ) _env.stem_darkening_cff=bool_val( s );
     s=getenv( "INFINALITY_FT_STEM_FITTING_STRENGTH" );
     if( s ) sscanf( s, "%d", &_env.stem_fitting_strength );
     s=getenv( "INFINALITY_FT_CHROMEOS_STYLE_SHARPENING_STRENGTH" );
diff -urN freetype-2.6.2.orig/src/base/ftinf.h freetype-2.6.2/src/base/ftinf.h
--- freetype-2.6.2.orig/src/base/ftinf.h	2015-12-07 12:42:05.976290812 +0000
+++ freetype-2.6.2/src/base/ftinf.h	2015-12-08 09:53:54.557327499 +0000
@@ -42,6 +42,8 @@
     int global_embolden_y_value;
     int grayscale_filter_strength;
     int stem_alignment_strength;
+    int stem_darkening_autofit;
+    int stem_darkening_cff;
     int stem_fitting_strength;
     int stem_snapping_sliding_scale;
     int use_known_settings_on_selected_fonts;
diff -urN freetype-2.6.2.orig/src/cff/cffobjs.c freetype-2.6.2/src/cff/cffobjs.c
--- freetype-2.6.2.orig/src/cff/cffobjs.c	2015-12-07 12:42:05.972957515 +0000
+++ freetype-2.6.2/src/cff/cffobjs.c	2015-12-08 09:56:23.342283975 +0000
@@ -34,6 +34,9 @@
 
 #include "cfferrs.h"
 
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+#include "../base/ftinf.h"
+#endif
 
   /*************************************************************************/
   /*                                                                       */
@@ -1053,6 +1056,9 @@
 #endif
 
     driver->no_stem_darkening = TRUE;
+#ifdef FT_CONFIG_OPTION_INFINALITY_PATCHSET
+    if(ftinf) driver->no_stem_darkening = !ftinf->stem_darkening_cff;
+#endif /* FT_CONFIG_OPTION_INFINALITY_PATCHSET */
 
     driver->darken_params[0] = CFF_CONFIG_OPTION_DARKENING_PARAMETER_X1;
     driver->darken_params[1] = CFF_CONFIG_OPTION_DARKENING_PARAMETER_Y1;
