From 5a661c0caa6960550f6903da7bc5b2712237e1d2 Mon Sep 17 00:00:00 2001
From: Sebastian Ramacher <sebastian@ramacher.at>
Date: Fri, 4 Nov 2016 15:58:32 +0100
Subject: [PATCH] Use sort and wildcard

Signed-off-by: Sebastian Ramacher <sebastian@ramacher.at>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 1fb42e2..835ebf7 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ include common.mk
 
 PROJECT  = zathura-pdf-mupdf
 PLUGIN   = pdf
-SOURCE   = $(shell find . -iname "*.c")
-HEADER   = $(shell find . -iname "*.h")
+SOURCE   = $(sort $(wildcard *.c))
+HEADER   = $(sort $(wildcard *.h))
 OBJECTS  = ${SOURCE:.c=.o}
 DOBJECTS = ${SOURCE:.c=.do}
 
