--- a/Makefile	2025-08-15 23:33:04.000000000 +0700
+++ b/Makefile	2025-08-21 13:06:01.707020236 +0700
@@ -16,7 +16,6 @@
 INSTALL_PROGRAM = $(INSTALL) -m 0755
 INSTALL_DATA    = $(INSTALL) -m 0644
 SO_SUFFIX = .so
-LIB_SUFFIX = .a
 
 # Make Configuration
 COMPILER_OPTIONS = -Wall -Wwrite-strings -Wcast-align -Wstrict-aliasing -Wstrict-prototypes -Wpointer-arith -Wundef
@@ -132,7 +131,7 @@
 OBJS += string.o time.o version.o win_unicode.o
 OBJS += $(ADDITIONAL_OBJECTS)
 
-all: sharedlib staticlib
+all: sharedlib
 	-@test "$(CROSS_DETECT)" = "cross" && echo "NOTICE: SIMD disabled: !x86_64 or a cross-compiler detected (CC = $(CC))" || true
 
 sharedlib: $(OBJS) $(SIMD_OBJS)
@@ -140,9 +139,6 @@
 	-test "$(ON_WINDOWS)" != "1" && $(LN) $(PROGRAM_NAME)$(SO_VER_FULL) $(PROGRAM_NAME)$(SO_VER_MAJOR)
 	-test "$(ON_WINDOWS)" != "1" && $(LN) $(PROGRAM_NAME)$(SO_VER_MAJOR) $(PROGRAM_NAME)$(SO_SUFFIX)
 
-staticlib: $(OBJS) $(SIMD_OBJS)
-	$(AR) rcs libjodycode$(LIB_SUFFIX) $(OBJS) $(SIMD_OBJS)
-
 jody_hash_simd.o:
 	$(CC) $(CFLAGS) $(COMPILER_OPTIONS) $(WIN_CFLAGS) $(CFLAGS_EXTRA) $(CPPFLAGS) -mavx2 -c -o jody_hash_simd.o jody_hash_simd.c
 
@@ -191,7 +187,6 @@
 	$(INSTALL_PROGRAM) $(PROGRAM_NAME)$(SO_VER_FULL) $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME)$(SO_VER_FULL)
 	-test "$(ON_WINDOWS)" != "1" && $(LN) $(PROGRAM_NAME)$(SO_VER_FULL) $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME)$(SO_VER_MAJOR)
 	-test "$(ON_WINDOWS)" != "1" && $(LN) $(PROGRAM_NAME)$(SO_VER_MAJOR) $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME)$(SO_SUFFIX)
-	$(INSTALL_DATA) $(PROGRAM_NAME)$(LIB_SUFFIX)  $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME)$(LIB_SUFFIX)
 	$(INSTALL_DATA) $(PROGRAM_NAME).h  $(DESTDIR)$(INC_DIR)/$(PROGRAM_NAME).h
 	$(INSTALL_DATA) $(PROGRAM_NAME).7  $(DESTDIR)$(MAN7_DIR)/$(PROGRAM_NAME).7
 
@@ -215,16 +209,14 @@
 test:
 	./test.sh
 
-stripped: sharedlib staticlib
+stripped: sharedlib
 	$(STRIP_UNNEEDED) libjodycode$(SO_SUFFIX)
-	$(STRIP_DEBUG) libjodycode$(LIB_SUFFIX)
 
 objsclean:
 	$(RM) $(OBJS) $(SIMD_OBJS) vercheck.o *.obj
 
 clean: objsclean
 	$(RM) $(PROGRAM_NAME)$(SO_SUFFIX) $(PROGRAM_NAME)$(SO_VER_MAJOR) $(PROGRAM_NAME)$(SO_VER_FULL)
-	$(RM) $(PROGRAM_NAME)$(LIB_SUFFIX) apiver cacheinfo vercheck
 	$(RM) *~ helper_code/*~ libjodycode.so.* libjodycode.dll.* .*.un~ *.gcno *.gcda *.gcov
 
 distclean: objsclean clean
