From c73ea16b3637c3bc3ece5e348f13d3bbc8fcb144 Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Thu, 18 Nov 2021 15:48:11 +0100
Subject: [PATCH 14/23] add header needed for IsGDKWaylandDisplay

---
 widget/gtk/WindowSurfaceProvider.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/widget/gtk/WindowSurfaceProvider.cpp b/widget/gtk/WindowSurfaceProvider.cpp
index a1183cbbe4f58..3b43e6ae71e3b 100644
--- a/widget/gtk/WindowSurfaceProvider.cpp
+++ b/widget/gtk/WindowSurfaceProvider.cpp
@@ -10,6 +10,7 @@
 #include "mozilla/gfx/Logging.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsWindow.h"
+#include "WidgetUtilsGtk.h"
 
 #ifdef MOZ_WAYLAND
 #  include "mozilla/StaticPrefs_widget.h"
-- 
2.34.1

