From 0a900c9553e77cd5a28c717c69cc1f92646fc356 Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Thu, 18 Nov 2021 15:53:43 +0100
Subject: [PATCH 16/23] always use EGL when building for wayland

---
 gfx/gl/moz.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gfx/gl/moz.build b/gfx/gl/moz.build
index 368490d06ba30..af395989ef34a 100644
--- a/gfx/gl/moz.build
+++ b/gfx/gl/moz.build
@@ -13,7 +13,7 @@ elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     gl_provider = "EAGL"
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "gtk":
-    if CONFIG["MOZ_EGL_XRENDER_COMPOSITE"]:
+    if CONFIG["MOZ_EGL_XRENDER_COMPOSITE"] or CONFIG["MOZ_WAYLAND"]:
         gl_provider = "EGL"
     else:
         gl_provider = "GLX"
-- 
2.34.1

