--- a/configure.ac
+++ b/configure.ac
@@ -43,7 +43,6 @@
 TORRENT_ENABLE_INTERRUPT_SOCKET

 TORRENT_ENABLE_ARCH
-TORRENT_WITH_SYSROOT

 dnl TORRENT_WITH_XFS
 TORRENT_WITHOUT_KQUEUE
@@ -61,7 +61,6 @@
 AX_PTHREAD

 PKG_CHECK_MODULES([ZLIB], [zlib])
-PKG_CHECK_MODULES([CPPUNIT], [cppunit],, [no_cppunit="yes"])

 CFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CFLAGS"
 CXXFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CXXFLAGS"
--- a/src/torrent/Makefile.am
+++ b/src/torrent/Makefile.am
@@ -58,8 +58,8 @@
        tracker_list.cc \
        tracker_list.h

-AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir)
-
+libsub_torrentutils_la_LIBADD = $(ZLIB_LIBS)
+AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir) $(ZLIB_CFLAGS)
 libtorrentincludedir = $(includedir)/torrent
 libtorrentinclude_HEADERS = \
 	bitfield.h \
