--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,6 @@
 TORRENT_ENABLE_INTERRUPT_SOCKET

 TORRENT_ENABLE_ARCH
-TORRENT_WITH_SYSROOT

 dnl TORRENT_WITH_XFS
 TORRENT_WITHOUT_KQUEUE
@@ -78,7 +77,7 @@
 AX_CHECK_ZLIB
 AX_PTHREAD

-PKG_CHECK_MODULES([CPPUNIT], [cppunit],, [no_cppunit="yes"])
+PKG_CHECK_MODULES([ZLIB], [zlib])

 CFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CFLAGS"
 CXXFLAGS="$PTHREAD_CFLAGS $CPPUNIT_CFLAGS $CXXFLAGS"
--- a/src/torrent/Makefile.am
+++ b/src/torrent//Makefile.am
@@ -136,8 +136,9 @@
        tracker_controller.h \
        tracker_list.cc \
        tracker_list.h
+libsub_torrentutils_la_LIBADD = $(ZLIB_LIBS)

-AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir)
+AM_CPPFLAGS = -I$(srcdir) -I$(srcdir)/.. -I$(top_srcdir) $(ZLIB_CFLAGS)

 libtorrent_torrent_data_includedir = $(includedir)/torrent/data
 libtorrent_torrent_data_include_HEADERS = \
