diff --git a/rbutil/rbutilqt/version.h b/rbutil/rbutilqt/version.h
index f0877ba..044c7c7 100644
--- a/version.h
+++ b/version.h
@@ -18,8 +18,6 @@
  *
  ****************************************************************************/

-#include "gitversion.h"
-
 // PUREVERSION is needed to be able to just compare versions. It does not
 // contain a build timestamp because it needs to be the same in different
 // files
@@ -40,7 +38,6 @@
 #define VERSIONSTRING(a, b, c) STR(a) "." STR(b) "." STR(c)
 #define VERSION VERSIONSTRING(VERSION_MAJOR, VERSION_MINOR, VERSION_MICRO) BUILDID
 // PUREVERSION should identify the build uniquely. Use version string for now.
-#define PUREVERSION GITHASH
-
-#define FULLVERSION VERSION " (" GITHASH "), built " __DATE__ " " __TIME__
+#define PUREVERSION "v1.4.1"

+#define FULLVERSION VERSION
