diff -Nuar xfsprogs-dev/include/Makefile xfsprogs-dev_new/include/Makefile
--- xfsprogs-dev/include/Makefile	2016-03-15 18:39:39.169459355 +0700
+++ xfsprogs-dev_new/include/Makefile	2016-03-15 18:37:41.000000000 +0700
@@ -46,7 +46,8 @@
 	$(PKG_PLATFORM).h \
 	xfs.h \
 	xqm.h \
-	xfs_arch.h
+	xfs_arch.h \
+	libxfs.h
 
 PHFILES = darwin.h freebsd.h irix.h linux.h gnukfreebsd.h
 LSRCFILES = $(shell echo $(PHFILES) | sed -e "s/$(PKG_PLATFORM).h//g")
