--- a/deluge/ui/web/js/deluge-all/TorrentGrid.js	2021-10-24 09:43:56.844470145 +0700
+++ b/deluge/ui/web/js/deluge-all/TorrentGrid.js	2021-10-24 09:45:24.768583485 +0700
@@ -126,7 +126,6 @@
             },
             {
                 header: _('Seeds'),
-                hidden: true,
                 width: 60,
                 sortable: true,
                 renderer: seedsRenderer,
@@ -134,7 +133,6 @@
             },
             {
                 header: _('Peers'),
-                hidden: true,
                 width: 60,
                 sortable: true,
                 renderer: peersRenderer,
@@ -179,7 +177,6 @@
             },
             {
                 header: _('Added'),
-                hidden: true,
                 width: 80,
                 sortable: true,
                 renderer: fdate,
@@ -203,7 +200,6 @@
             },
             {
                 header: _('Tracker'),
-                hidden: true,
                 width: 120,
                 sortable: true,
                 renderer: trackerRenderer,
@@ -211,7 +207,6 @@
             },
             {
                 header: _('Download Folder'),
-                hidden: true,
                 width: 120,
                 sortable: true,
                 renderer: fplain,
@@ -219,6 +214,7 @@
             },
             {
                 header: _('Owner'),
+                hidden: true,
                 width: 80,
                 sortable: true,
                 renderer: fplain,
