--- wscript	2017-10-31 15:27:25.100037571 +0700
+++ wscript_	2017-10-31 18:23:58.912542575 +0700
@@ -426,12 +426,12 @@
 ]
 
 ffmpeg_pkg_config_checks = [
-    'libavutil',     '>= 56.0.100',
-    'libavcodec',    '>= 58.2.100',
-    'libavformat',   '>= 58.0.102',
-    'libswscale',    '>= 5.0.101',
-    'libavfilter',   '>= 7.0.101',
-    'libswresample', '>= 3.0.100',
+    'libavutil',     '>= 55.0.2',
+    'libavcodec',    '>= 56.0.1',
+    'libavformat',   '>= 56.0.1',
+    'libswscale',    '>= 4.0.1',
+    'libavfilter',   '>= 6.0.1',
+    'libswresample', '>= 2.0.1',
 ]
 libav_pkg_config_checks = [
     'libavutil',     '>= 56.6.0',
@@ -484,7 +484,7 @@
     }, {
         'name': 'libav-any',
         'desc': 'Libav/FFmpeg library versions',
-        'deps': 'ffmpeg_mpv || libav',
+        'deps': 'if_ffmpeg || libav',
         'func': check_ffmpeg_or_libav_versions(),
         'req': True,
         'fmsg': "Unable to find development files for some of the required \
