Index: chromium-74.0.3729.108/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc
===================================================================
--- chromium-74.0.3729.108.orig/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc
+++ chromium-74.0.3729.108/third_party/webrtc/modules/desktop_capture/linux/base_capturer_pipewire.cc
@@ -20,6 +20,7 @@
 
 #include <memory>
 #include <utility>
+#include <cstring>
 
 #include "absl/memory/memory.h"
 #include "modules/desktop_capture/desktop_capture_options.h"
