--- a/meson.build	2018-06-07 07:54:35.253030701 +0200
+++ b/meson.build	2018-06-07 07:56:25.981031567 +0200
@@ -2927,20 +2927,7 @@
 # install_data('modprobe.d/systemd.conf',
 #              install_dir : modprobedir)
 #endif // 0
-install_data('LICENSE.GPL2',
-             'LICENSE.LGPL2.1',
-             'NEWS',
-             'README',
-             'doc/CODING_STYLE',
-#if 0 /// irrelevant for elogind
-#              'doc/DISTRO_PORTING',
-#              'doc/ENVIRONMENT.md',
-#              'doc/HACKING',
-#              'doc/TRANSIENT-SETTINGS.md',
-#              'doc/TRANSLATORS',
-#              'doc/UIDS-GIDS.md',
-#endif // 0
-             'src/libelogind/sd-bus/GVARIANT-SERIALIZATION',
+install_data('src/libelogind/sd-bus/GVARIANT-SERIALIZATION',
              install_dir : docdir)
 
 #if 0 /// UNNEEDED by elogind
