--- a/meson.build	2018-04-10 13:59:30.950795634 +0700
+++ b/meson.build	2018-04-10 14:00:19.504391757 +0700
@@ -2800,21 +2800,6 @@
 # install_data('modprobe.d/systemd.conf',
 #              install_dir : modprobedir)
 #endif // 0
-install_data('README',
-             'NEWS',
-             'CODING_STYLE',
-#if 0 /// UNNEEDED by elogind
-#              'DISTRO_PORTING',
-#              'ENVIRONMENT.md',
-#endif // 0
-             'LICENSE.GPL2',
-             'LICENSE.LGPL2.1',
-             'src/libelogind/sd-bus/GVARIANT-SERIALIZATION',
-#if 0 /// UNNEEDED by elogind
-#              'TRANSIENT-SETTINGS.md',
-#              'UIDS-GIDS.md',
-#endif // 0
-             install_dir : docdir)
 
 #if 0 /// UNNEEDED by elogind
 # meson.add_install_script('sh', '-c', mkdir_p.format(systemdstatedir))
