--- a/programs/Makefile	2015-10-24 22:52:40.370336275 +0700
+++ b/programs/Makefile	2015-10-24 22:53:19.895828564 +0700
@@ -238,7 +238,7 @@
 	./datagen -g256MB | ./lz4 -vqB4D | ./lz4 -t
 	./datagen -g6GB   | ./lz4 -vqB5D | ./lz4 -qt
 	./datagen -g6GB   | ./lz4 -vq9BD | ./lz4 -qt
-	@rm tmp*
+	@rm -f tmp*
 
 test-lz4: lz4 datagen test-lz4-basic test-lz4-multiple test-lz4-sparse test-lz4-contentSize test-lz4-frame-concatenation
 	@echo "\n ---- test pass-through ----"
