diff -ur qBittorrent-release-3.3.1.orig/unixconf.pri qBittorrent-release-3.3.1/unixconf.pri
--- qBittorrent-release-3.3.1.orig/unixconf.pri	2015-12-08 12:22:28.916178783 +0100
+++ qBittorrent-release-3.3.1/unixconf.pri	2015-12-08 13:15:19.582896405 +0100
@@ -15,8 +15,8 @@
 # COMPILATION SPECIFIC
 !nogui:dbus: QT += dbus
 
-QMAKE_CXXFLAGS += -Wformat -Wformat-security
-!haiku: QMAKE_LFLAGS_APP += -rdynamic
+QMAKE_CXXFLAGS += -fPIE -Wformat -Wformat-security
+!haiku: QMAKE_LFLAGS_APP += -Wl,-z,now -pie -rdynamic
 
 # Man page
 nogui {
