diff --git a/mail/installer/package-manifest.in b/mail/installer/package-manifest.in
--- a/mail/installer/package-manifest.in
+++ b/mail/installer/package-manifest.in
@@ -329,16 +329,17 @@
 @RESPATH@/components/mozldap.xpt
 @RESPATH@/components/nsAbLDAPAttributeMap.js
 @RESPATH@/components/nsAbLDAPAutoCompleteSearch.js
 @RESPATH@/components/nsLDAPProtocolHandler.js
 @RESPATH@/components/ldapComponents.manifest
 @BINPATH@/@DLL_PREFIX@ldap60@DLL_SUFFIX@
 @BINPATH@/@DLL_PREFIX@ldif60@DLL_SUFFIX@
 @BINPATH@/@DLL_PREFIX@prldap60@DLL_SUFFIX@
+@BINPATH@/@DLL_PREFIX@ssldap60@DLL_SUFFIX@
 
 ; download progress for jsdownloads
 @RESPATH@/components/DownloadsStartup.js
 
 ; Protocol/Content handling
 @RESPATH@/components/nsSMTPProtocolHandler.js
 @RESPATH@/components/nsSMTPProtocolHandler.manifest
 
