--- Python-3.6.8/Makefile.pre.in
+++ Python-3.6.8/Makefile.pre.in
@@ -1103,6 +1103,18 @@
 		$(RUNSHARED) $(PYTHON_FOR_BUILD) -m ensurepip \
 			$$ensurepip --root=$(DESTDIR)/ ; \
 	fi
+	@if test -f $(LIBRARY) && test $(LIBRARY) != $(LDLIBRARY); then \
+		if test "$(PYTHONFRAMEWORKDIR)" = no-framework; then \
+			if test "$(SHLIB_SUFFIX)" = .dll; then \
+				$(INSTALL_DATA) $(LDLIBRARY) $(DESTDIR)$(LIBPL); \
+			else \
+				$(INSTALL_DATA) $(LIBRARY) $(DESTDIR)$(LIBPL)/$(LIBRARY); \
+			fi; \
+		else \
+			echo "Skipped install of $(LIBRARY) - use make frameworkinstall"; \
+		fi; \
+	else	true; \
+	fi
 
 commoninstall:  check-clean-src @FRAMEWORKALTINSTALLFIRST@ \
 		altbininstall libinstall inclinstall libainstall \
@@ -1428,17 +1440,6 @@
 		else	true; \
 		fi; \
 	done
-	@if test -d $(LIBRARY); then :; else \
-		if test "$(PYTHONFRAMEWORKDIR)" = no-framework; then \
-			if test "$(SHLIB_SUFFIX)" = .dll; then \
-				$(INSTALL_DATA) $(LDLIBRARY) $(DESTDIR)$(LIBPL) ; \
-			else \
-				$(INSTALL_DATA) $(LIBRARY) $(DESTDIR)$(LIBPL)/$(LIBRARY) ; \
-			fi; \
-		else \
-			echo Skip install of $(LIBRARY) - use make frameworkinstall; \
-		fi; \
-	fi
 	$(INSTALL_DATA) Modules/config.c $(DESTDIR)$(LIBPL)/config.c
 	$(INSTALL_DATA) Programs/python.o $(DESTDIR)$(LIBPL)/python.o
 	$(INSTALL_DATA) $(srcdir)/Modules/config.c.in $(DESTDIR)$(LIBPL)/config.c.in
--- Python-3.6.8/Misc/python-config.in
+++ Python-3.6.8/Misc/python-config.in
@@ -50,11 +50,7 @@
         libs = ['-lpython' + pyver + sys.abiflags]
         libs += getvar('LIBS').split()
         libs += getvar('SYSLIBS').split()
-        # add the prefix/lib/pythonX.Y/config dir, but only if there is no
-        # shared library in prefix/lib/.
         if opt == '--ldflags':
-            if not getvar('Py_ENABLE_SHARED'):
-                libs.insert(0, '-L' + getvar('LIBPL'))
             if not getvar('PYTHONFRAMEWORK'):
                 libs.extend(getvar('LINKFORSHARED').split())
         print(' '.join(libs))
--- Python-3.6.8/Misc/python-config.sh.in
+++ Python-3.6.8/Misc/python-config.sh.in
@@ -46,7 +46,6 @@
 LDLIBRARY="@LDLIBRARY@"
 LINKFORSHARED="@LINKFORSHARED@"
 OPT="@OPT@"
-PY_ENABLE_SHARED="@PY_ENABLE_SHARED@"
 LDVERSION="@LDVERSION@"
 LIBDEST=${prefix_real}/lib/python${VERSION}
 LIBPL=$(echo "@LIBPL@" | sed "s#$prefix#$prefix_real#")
@@ -93,11 +92,7 @@
             if [ -z "$PYTHONFRAMEWORK" ] ; then
                 LINKFORSHAREDUSED=$LINKFORSHARED
             fi
-            LIBPLUSED=
-            if [ "$PY_ENABLE_SHARED" = "0" ] ; then
-                LIBPLUSED="-L$LIBPL"
-            fi
-            echo "$LIBPLUSED -L$libdir $LIBS $LINKFORSHAREDUSED"
+            echo "-L$libdir $LIBS $LINKFORSHAREDUSED"
         ;;
         --extension-suffix)
             echo "$SO"
--- Python-3.6.8/Modules/makesetup
+++ Python-3.6.8/Modules/makesetup
@@ -90,7 +90,7 @@
 	 then
 	 	ExtraLibDir=.
 	 else
-	 	ExtraLibDir='$(LIBPL)'
+	 	ExtraLibDir='$(LIBDIR)'
 	 fi
 	 ExtraLibs="-L$ExtraLibDir -lpython\$(VERSION)";;
 esac
