Index: VirtualBox-6.1.4/src/VBox/Debugger/DBGCDumpImage.cpp
===================================================================
--- VirtualBox-6.1.4.orig/src/VBox/Debugger/DBGCDumpImage.cpp
+++ VirtualBox-6.1.4/src/VBox/Debugger/DBGCDumpImage.cpp
@@ -468,7 +468,7 @@ static const char *dbgcMachoCpuType(uint
 {
     switch (uType)
     {
-        case CPU_TYPE_ANY:          return "CPU_TYPE_ANY";
+        default:		    return "CPU_TYPE_ANY";
         case CPU_TYPE_VAX:          return "VAX";
         case CPU_TYPE_MC680x0:      return "MC680x0";
         case CPU_TYPE_X86:          return "X86";
