Index: VirtualBox-5.2.14/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
===================================================================
--- VirtualBox-5.2.14.orig/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
+++ VirtualBox-5.2.14/src/VBox/Frontends/VirtualBox/src/precomp_vcc.h
@@ -83,6 +83,7 @@
 #include <QtWidgets/qgraphicsitem.h>
 #include <QtWidgets/qgraphicslayoutitem.h>
 #include <QtWidgets/qgraphicswidget.h>
+#include <QtWidgets/qbuttongroup.h>
 #include <QtCore/QMetaType>
 #include <QtGui/qevent.h>
 #include <QtGui/qtouchdevice.h>
Index: VirtualBox-5.2.14/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
===================================================================
--- VirtualBox-5.2.14.orig/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
+++ VirtualBox-5.2.14/src/VBox/Frontends/VirtualBox/src/settings/global/UIGlobalSettingsProxy.cpp
@@ -22,6 +22,7 @@
 /* Qt includes: */
 # include <QButtonGroup>
 # include <QRegExpValidator>
+#include <QButtonGroup>
 
 /* GUI includes: */
 # include "QIWidgetValidator.h"
