--- a/build/pgo/profileserver.py	2019-09-30 20:58:31.169527743 +0700
+++ b/build/pgo/profileserver.py	2019-09-30 21:02:07.964185254 +0700
@@ -132,7 +132,7 @@
         env["LLVM_PROFILE_FILE"] = "default_%p_random_%m.profraw"
 
         # Write to an output file if we're running in automation
-        process_args = {}
+        process_args = {'universal_newlines': True}
         if 'UPLOAD_PATH' in env:
             process_args['logfile'] = os.path.join(env['UPLOAD_PATH'], 'profile-run-1.log')
@@ -137,7 +137,6 @@
                                env=env,
                                process_args=process_args)
         runner.start(debug_args=debug_args, interactive=interactive)
-        ret = runner.wait()
         httpd.stop()
         if ret:
             print("Firefox exited with code %d during profiling" % ret)
