From 7276e1c36841f65b7640b9982a1c850da7918c3a Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Thu, 18 Nov 2021 15:53:43 +0100
Subject: [PATCH 14/18] always use EGL when building for wayland

---
 gfx/gl/moz.build | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/gfx/gl/moz.build b/gfx/gl/moz.build
index 03434957ccb51..c4727d79a1c3e 100644
--- a/gfx/gl/moz.build
+++ b/gfx/gl/moz.build
@@ -13,10 +13,10 @@ elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "cocoa":
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "uikit":
     gl_provider = "EAGL"
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "gtk":
-    if CONFIG["MOZ_EGL_XRENDER_COMPOSITE"]:
-        gl_provider = "EGL"
-    else:
+    if CONFIG["MOZ_X11"]:
         gl_provider = "GLX"
+    else:
+        gl_provider = "EGL"
 elif CONFIG["MOZ_WIDGET_TOOLKIT"] == "android":
     gl_provider = "EGL"
 
@@ -117,7 +117,10 @@ elif gl_provider == "GLX":
     EXPORTS += ["GLContextGLX.h", "GLXLibrary.h"]
 
 if CONFIG["MOZ_WAYLAND"]:
-    SOURCES += ["GLContextProviderWayland.cpp", "SharedSurfaceDMABUF.cpp"]
+    SOURCES += ["SharedSurfaceDMABUF.cpp"]
+
+if CONFIG["MOZ_X11"] and CONFIG["MOZ_WAYLAND"]:
+    SOURCES += ["GLContextProviderWayland.cpp"]
 
 UNIFIED_SOURCES += [
     "AndroidSurfaceTexture.cpp",
-- 
2.35.1

