Index: chromium-73.0.3683.75/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
===================================================================
--- chromium-73.0.3683.75.orig/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
+++ chromium-73.0.3683.75/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
@@ -45,6 +45,9 @@ static void ReportToUMA(VAJDADecoderFail
 // macro in the header file without pulling in the entire <va/va.h>.
 static_assert(VA_FOURCC_YUYV == VA_FOURCC('Y', 'U', 'Y', 'V'),
               "VA_FOURCC_YUYV must be equal to VA_FOURCC('Y', 'U', 'Y', 'V')");
+#ifndef VA_FOURCC_I420
+#define VA_FOURCC_I420 0x30323449
+#endif
 constexpr VAImageFormat kImageFormatI420 = {.fourcc = VA_FOURCC_I420,
                                             .byte_order = VA_LSB_FIRST,
                                             .bits_per_pixel = 12};
Index: chromium-73.0.3683.75/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- chromium-73.0.3683.75.orig/media/gpu/vaapi/vaapi_wrapper.cc
+++ chromium-73.0.3683.75/media/gpu/vaapi/vaapi_wrapper.cc
@@ -278,9 +278,6 @@ bool VADisplayState::Initialize() {
 }
 
 bool VADisplayState::InitializeOnce() {
-  static_assert(VA_MAJOR_VERSION >= 1 && VA_MINOR_VERSION >= 1,
-                "Requires VA-API >= 1.1.0");
-
   switch (gl::GetGLImplementation()) {
     case gl::kGLImplementationEGLGLES2:
       va_display_ = vaGetDisplayDRM(drm_fd_.get());
