Index: chromium-73.0.3673.0/gpu/config/gpu_info.cc
===================================================================
--- chromium-73.0.3673.0.orig/gpu/config/gpu_info.cc
+++ chromium-73.0.3673.0/gpu/config/gpu_info.cc
@@ -112,15 +112,15 @@ GPUInfo::GPUDevice::GPUDevice()
 
 GPUInfo::GPUDevice::GPUDevice(const GPUInfo::GPUDevice& other) = default;
 
-GPUInfo::GPUDevice::GPUDevice(GPUInfo::GPUDevice&& other) noexcept = default;
+GPUInfo::GPUDevice::GPUDevice(GPUInfo::GPUDevice&& other) = default;
 
-GPUInfo::GPUDevice::~GPUDevice() noexcept = default;
+GPUInfo::GPUDevice::~GPUDevice() = default;
 
 GPUInfo::GPUDevice& GPUInfo::GPUDevice::operator=(
     const GPUInfo::GPUDevice& other) = default;
 
 GPUInfo::GPUDevice& GPUInfo::GPUDevice::operator=(
-    GPUInfo::GPUDevice&& other) noexcept = default;
+    GPUInfo::GPUDevice&& other) = default;
 
 GPUInfo::GPUInfo()
     : optimus(false),
Index: chromium-73.0.3673.0/components/policy/core/common/policy_map.cc
===================================================================
--- chromium-73.0.3673.0.orig/components/policy/core/common/policy_map.cc
+++ chromium-73.0.3673.0/components/policy/core/common/policy_map.cc
@@ -17,8 +17,8 @@ PolicyMap::Entry::Entry() = default;
 
 PolicyMap::Entry::~Entry() = default;
 
-PolicyMap::Entry::Entry(Entry&&) noexcept = default;
-PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) noexcept = default;
+PolicyMap::Entry::Entry(Entry&&) = default;
+PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) = default;
 
 PolicyMap::Entry PolicyMap::Entry::DeepCopy() const {
   Entry copy;
Index: chromium-73.0.3673.0/components/search_provider_logos/logo_common.cc
===================================================================
--- chromium-73.0.3673.0.orig/components/search_provider_logos/logo_common.cc
+++ chromium-73.0.3673.0/components/search_provider_logos/logo_common.cc
@@ -12,24 +12,24 @@ const int64_t kMaxTimeToLiveMS = INT64_C
 
 LogoMetadata::LogoMetadata() = default;
 LogoMetadata::LogoMetadata(const LogoMetadata&) = default;
-LogoMetadata::LogoMetadata(LogoMetadata&&) noexcept = default;
+LogoMetadata::LogoMetadata(LogoMetadata&&) = default;
 LogoMetadata& LogoMetadata::operator=(const LogoMetadata&) = default;
-LogoMetadata& LogoMetadata::operator=(LogoMetadata&&) noexcept = default;
+LogoMetadata& LogoMetadata::operator=(LogoMetadata&&) = default;
 LogoMetadata::~LogoMetadata() = default;
 
 EncodedLogo::EncodedLogo() = default;
 EncodedLogo::EncodedLogo(const EncodedLogo&) = default;
-EncodedLogo::EncodedLogo(EncodedLogo&&) noexcept = default;
+EncodedLogo::EncodedLogo(EncodedLogo&&) = default;
 EncodedLogo& EncodedLogo::operator=(const EncodedLogo&) = default;
-EncodedLogo& EncodedLogo::operator=(EncodedLogo&&) noexcept = default;
+EncodedLogo& EncodedLogo::operator=(EncodedLogo&&) = default;
 EncodedLogo::~EncodedLogo() = default;
 
 Logo::Logo() = default;
 Logo::~Logo() = default;
 
 LogoCallbacks::LogoCallbacks() = default;
-LogoCallbacks::LogoCallbacks(LogoCallbacks&&) noexcept = default;
-LogoCallbacks& LogoCallbacks::operator=(LogoCallbacks&&) noexcept = default;
+LogoCallbacks::LogoCallbacks(LogoCallbacks&&) = default;
+LogoCallbacks& LogoCallbacks::operator=(LogoCallbacks&&) = default;
 LogoCallbacks::~LogoCallbacks() = default;
 
 }  // namespace search_provider_logos
Index: chromium-73.0.3673.0/components/signin/core/browser/account_info.cc
===================================================================
--- chromium-73.0.3673.0.orig/components/signin/core/browser/account_info.cc
+++ chromium-73.0.3673.0/components/signin/core/browser/account_info.cc
@@ -28,11 +28,11 @@ AccountInfo::~AccountInfo() = default;
 
 AccountInfo::AccountInfo(const AccountInfo& other) = default;
 
-AccountInfo::AccountInfo(AccountInfo&& other) noexcept = default;
+AccountInfo::AccountInfo(AccountInfo&& other) = default;
 
 AccountInfo& AccountInfo::operator=(const AccountInfo& other) = default;
 
-AccountInfo& AccountInfo::operator=(AccountInfo&& other) noexcept = default;
+AccountInfo& AccountInfo::operator=(AccountInfo&& other) = default;
 
 bool AccountInfo::IsEmpty() const {
   return account_id.empty() && email.empty() && gaia.empty() &&
Index: chromium-73.0.3673.0/chrome/common/media_router/media_sink.cc
===================================================================
--- chromium-73.0.3673.0.orig/chrome/common/media_router/media_sink.cc
+++ chromium-73.0.3673.0/chrome/common/media_router/media_sink.cc
@@ -19,12 +19,12 @@ MediaSink::MediaSink(const MediaSink::Id
       provider_id_(provider_id) {}
 
 MediaSink::MediaSink(const MediaSink& other) = default;
-MediaSink::MediaSink(MediaSink&& other) noexcept = default;
+MediaSink::MediaSink(MediaSink&& other) = default;
 MediaSink::MediaSink() = default;
 MediaSink::~MediaSink() = default;
 
 MediaSink& MediaSink::operator=(const MediaSink& other) = default;
-MediaSink& MediaSink::operator=(MediaSink&& other) noexcept = default;
+MediaSink& MediaSink::operator=(MediaSink&& other) = default;
 
 bool MediaSink::IsMaybeCloudSink() const {
   switch (icon_type_) {
