From e2925653686b4a209140a1808b744ee61157d44a Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Thu, 18 Nov 2021 15:46:17 +0100
Subject: [PATCH 09/18] enable glxtest on X11 and wayland

---
 toolkit/xre/moz.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/toolkit/xre/moz.build b/toolkit/xre/moz.build
index b1e2b58621d64..576ec91669798 100644
--- a/toolkit/xre/moz.build
+++ b/toolkit/xre/moz.build
@@ -156,7 +156,7 @@ SOURCES += [
     "ProfileReset.cpp",
 ]
 
-if CONFIG["MOZ_X11"]:
+if CONFIG["MOZ_X11"] or CONFIG["MOZ_WAYLAND"]:
     UNIFIED_SOURCES += [
         "glxtest.cpp",
     ]
@@ -187,7 +187,7 @@ include("/ipc/chromium/chromium-config.mozbuild")
 
 FINAL_LIBRARY = "xul"
 
-if CONFIG["MOZ_X11"]:
+if CONFIG["MOZ_X11"] or CONFIG["MOZ_WAYLAND"]:
     DEFINES["USE_GLX_TEST"] = True
 
 for var in (
-- 
2.35.1

