From 383039d3434f2549c0119e67db65f12cc9f34b6b Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Thu, 18 Nov 2021 15:48:11 +0100
Subject: [PATCH 14/23] add header needed for IsGDKWaylandDisplay

---
 widget/gtk/WindowSurfaceProvider.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/widget/gtk/WindowSurfaceProvider.cpp b/widget/gtk/WindowSurfaceProvider.cpp
index 403af8e445bc8..94ba1be0ff171 100644
--- a/widget/gtk/WindowSurfaceProvider.cpp
+++ b/widget/gtk/WindowSurfaceProvider.cpp
@@ -10,6 +10,7 @@
 #include "mozilla/gfx/Logging.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsWindow.h"
+#include "WidgetUtilsGtk.h"
 
 #ifdef MOZ_WAYLAND
 #  include "mozilla/StaticPrefs_widget.h"
-- 
2.35.1

