From bf0c824817fce422d1e927ac1554100ed1c07397 Mon Sep 17 00:00:00 2001
From: Tilmann Meyer <me@atiltedtree.dev>
Date: Mon, 13 Dec 2021 22:43:55 +0100
Subject: [PATCH 21/23] disable FFVPX

---
 toolkit/moz.configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index c3793ec1756a4..7725dbef45e89 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1987,7 +1987,7 @@ with only_when(compile_environment):
 
     @depends(target)
     def libav_fft(target):
-        return target.kernel == "WINNT" or target.cpu == "x86_64"
+        return False
 
     set_config("MOZ_LIBAV_FFT", depends(when=libav_fft)(lambda: True))
     set_define("MOZ_LIBAV_FFT", depends(when=libav_fft)(lambda: True))
@@ -1998,7 +1998,7 @@ with only_when(compile_environment | artifact_builds):
 
     @depends(target)
     def ffvpx(target):
-        enable = use_nasm = True
+        enable = use_nasm = False
         flac_only = False
         flags = []
 
-- 
2.35.1

