--- a/browser/moz.configure	2017-07-31 23:20:49.000000000 +0700
+++ b/browser/moz.configure	2017-09-19 12:57:57.228354679 +0700
@@ -5,7 +5,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 imply_option('MOZ_PLACES', True)
-imply_option('MOZ_SERVICES_HEALTHREPORT', True)
+imply_option('MOZ_SERVICES_HEALTHREPORT', False)
 imply_option('MOZ_SERVICES_SYNC', True)
 
 include('../toolkit/moz.configure')
