--- moz.build	2017-09-15 23:14:47.000000000 +0700
+++ moz.build_new	2017-09-29 18:32:33.068758645 +0700
@@ -5,17 +5,8 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DIRS += [
-    'activity-stream',
-    'aushelper',
-    'clicktoplay-rollout',
     'e10srollout',
-    'followonsearch',
-    'formautofill',
-    'onboarding',
     'pdfjs',
-    'pocket',
-    'screenshots',
-    'shield-recipe-client',
     'webcompat',
 ]
 
