From: Emilio Pozuelo Monfort <pochu@debian.org>
Date: Sat, 7 Dec 2019 08:55:17 +0900
Subject: Don't build ICU in parallel

---
 intl/icu_sources_data.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/intl/icu_sources_data.py b/intl/icu_sources_data.py
index e366f056c65..c6d075da5a0 100644
--- a/intl/icu_sources_data.py
+++ b/intl/icu_sources_data.py
@@ -244,9 +244,7 @@ def update_data_file(topsrcdir):
     print('Running ICU make...')
     if not try_run(
             'icu-make',
-            ['make',
-             '--jobs=%d' % multiprocessing.cpu_count(),
-             '--output-sync'],
+            ['make'],
             cwd=objdir):
         return False
     print('Copying ICU data file...')
