diff --git a/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h b/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
--- a/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
+++ b/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
@@ -38,14 +38,6 @@
     if (aParams.VideoConfig().HasAlpha()) {
       return nullptr;
     }
-    if (VPXDecoder::IsVPX(aParams.mConfig.mMimeType) &&
-        aParams.mOptions.contains(CreateDecoderParams::Option::LowLatency) &&
-        !StaticPrefs::media_ffmpeg_low_latency_enabled()) {
-      // We refuse to create a decoder with low latency enabled if it's VP8 or
-      // VP9 unless specifically allowed: this will fallback to libvpx later.
-      // We do allow it for h264.
-      return nullptr;
-    }
     RefPtr<MediaDataDecoder> decoder = new FFmpegVideoDecoder<V>(
         mLib, aParams.VideoConfig(), aParams.mKnowsCompositor,
         aParams.mImageContainer,
diff --git a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
--- a/modules/libpref/init/StaticPrefList.yaml
+++ b/modules/libpref/init/StaticPrefList.yaml
@@ -7109,13 +7109,6 @@
     mirror: always
 #endif
 
-#if defined(MOZ_FFMPEG) || defined(MOZ_FFVPX)
--   name: media.ffmpeg.low-latency.enabled
-    type: RelaxedAtomicBool
-    value: false
-    mirror: always
-#endif
-
 #ifdef MOZ_WMF
 
 -   name: media.wmf.enabled

