From 3a8367663fbe400c135f2ad01261708f1d2d5ca3 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 30 Nov 2017 15:47:34 +0100
Subject: [PATCH 02/64] build: DEPS: pick an existing webrtc commit

*sigh*
---
 DEPS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DEPS b/DEPS
index b404d70c090d..ff0ae67141b4 100644
--- a/DEPS
+++ b/DEPS
@@ -176,7 +176,7 @@ deps = {
   'src/third_party/webgl/src':
     (Var("chromium_git")) + '/external/khronosgroup/webgl.git@559e304b3903548b4f8e0681ba3b1030aa0d29d9',
   'src/third_party/webrtc':
-    (Var("chromium_git")) + '/external/webrtc/trunk/webrtc.git@a2329acf18dc0653106782f63f54675dab9c483e',
+    (Var("chromium_git")) + '/external/webrtc/trunk/webrtc.git@59d92ff1288817c52b13f54871ea42619231cf7a',
   'src/third_party/yasm/source/patched-yasm':
     (Var("chromium_git")) + '/chromium/deps/yasm/patched-yasm.git@b98114e18d8b9b84586b10d24353ab8616d4c5fc',
   'src/tools/gyp':
-- 
2.15.0

