diff -up firefox-81.0/dom/media/platforms/PDMFactory.cpp.firefox-disable-ffvpx-with-vapi firefox-81.0/dom/media/platforms/PDMFactory.cpp
--- firefox-81.0/dom/media/platforms/PDMFactory.cpp.firefox-disable-ffvpx-with-vapi	2020-09-17 02:32:43.000000000 +0200
+++ firefox-81.0/dom/media/platforms/PDMFactory.cpp	2020-09-21 10:30:29.393903183 +0200
@@ -371,12 +371,6 @@ void PDMFactory::CreatePDMs() {
     StartupPDM(m);
   }
 #endif
-#ifdef MOZ_FFVPX
-  if (StaticPrefs::media_ffvpx_enabled()) {
-    m = FFVPXRuntimeLinker::CreateDecoderModule();
-    StartupPDM(m);
-  }
-#endif
 #ifdef MOZ_FFMPEG
   if (StaticPrefs::media_ffmpeg_enabled()) {
     m = FFmpegRuntimeLinker::CreateDecoderModule();
@@ -385,6 +379,12 @@ void PDMFactory::CreatePDMs() {
     mFFmpegFailedToLoad = false;
   }
 #endif
+#ifdef MOZ_FFVPX
+  if (StaticPrefs::media_ffvpx_enabled()) {
+    m = FFVPXRuntimeLinker::CreateDecoderModule();
+    StartupPDM(m);
+  }
+#endif
 #ifdef MOZ_WIDGET_ANDROID
   if (StaticPrefs::media_android_media_codec_enabled()) {
     m = new AndroidDecoderModule();
