diff --git chromium-71.0.3578.30/base/bind_internal.h chromium-71.0.3578.30/base/bind_internal.h
index aff03a6347..65f9590acc 100644
--- chromium-71.0.3578.30/base/bind_internal.h
+++ chromium-71.0.3578.30/base/bind_internal.h
@@ -749,6 +749,7 @@ bool QueryCancellationTraitsImpl(BindStateBase::CancellationQueryMode mode,
           functor, std::get<indices>(bound_args)...);
   }
   NOTREACHED();
+  __builtin_abort ();
 }
 
 // Relays |base| to corresponding CallbackCancellationTraits<>::Run(). Returns
diff --git chromium-71.0.3578.30/base/callback_internal.cc chromium-71.0.3578.30/base/callback_internal.cc
index 6a185d9092..f0f4157783 100644
--- chromium-71.0.3578.30/base/callback_internal.cc
+++ chromium-71.0.3578.30/base/callback_internal.cc
@@ -21,6 +21,7 @@ bool QueryCancellationTraitsForNonCancellables(
       return true;
   }
   NOTREACHED();
+  __builtin_abort ();
 }
 
 }  // namespace
diff --git chromium-71.0.3578.30/cc/input/snap_fling_controller.cc chromium-71.0.3578.30/cc/input/snap_fling_controller.cc
index 4c8e485bcf..00764506d9 100644
--- chromium-71.0.3578.30/cc/input/snap_fling_controller.cc
+++ chromium-71.0.3578.30/cc/input/snap_fling_controller.cc
@@ -26,6 +26,8 @@ bool SnapFlingController::FilterEventForSnap(
     case GestureScrollType::kEnd: {
       return state_ == State::kActive || state_ == State::kFinished;
     }
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/background_fetch/background_fetch_download_client.cc chromium-71.0.3578.30/chrome/browser/background_fetch/background_fetch_download_client.cc
index 27cbf6d17f..838cdbc466 100644
--- chromium-71.0.3578.30/chrome/browser/background_fetch/background_fetch_download_client.cc
+++ chromium-71.0.3578.30/chrome/browser/background_fetch/background_fetch_download_client.cc
@@ -36,6 +36,8 @@ BackgroundFetchFailureReason ToBackgroundFetchFailureReason(
     case download::Client::FailureReason::ABORTED:
     case download::Client::FailureReason::CANCELLED:
       return BackgroundFetchFailureReason::CANCELLED;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/chrome_browser_main.cc chromium-71.0.3578.30/chrome/browser/chrome_browser_main.cc
index 3be69bf7da..4d5530ffac 100644
--- chromium-71.0.3578.30/chrome/browser/chrome_browser_main.cc
+++ chromium-71.0.3578.30/chrome/browser/chrome_browser_main.cc
@@ -746,6 +746,8 @@ bool WaitUntilMachineLevelUserCloudPolicyEnrollmentFinished(
     case RegisterResult::kQuitDueToFailure:
       chrome::AttemptExit();
       return false;
+    default:
+      __builtin_abort ();
   }
 #else
   return true;
diff --git chromium-71.0.3578.30/chrome/browser/download/offline_item_model.cc chromium-71.0.3578.30/chrome/browser/download/offline_item_model.cc
index a7387fed4e..3d1177c1d7 100644
--- chromium-71.0.3578.30/chrome/browser/download/offline_item_model.cc
+++ chromium-71.0.3578.30/chrome/browser/download/offline_item_model.cc
@@ -147,6 +147,8 @@ download::DownloadItem::DownloadState OfflineItemModel::GetState() const {
     case OfflineItemState::MAX_DOWNLOAD_STATE:
       NOTREACHED();
       return download::DownloadItem::CANCELLED;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/download/offline_item_utils.cc chromium-71.0.3578.30/chrome/browser/download/offline_item_utils.cc
index 7a1fb4a4b8..299247c189 100644
--- chromium-71.0.3578.30/chrome/browser/download/offline_item_utils.cc
+++ chromium-71.0.3578.30/chrome/browser/download/offline_item_utils.cc
@@ -199,6 +199,8 @@ FailState OfflineItemUtils::ConvertDownloadInterruptReasonToFailState(
       return FailState::CRASH;
     case download::DOWNLOAD_INTERRUPT_REASON_NONE:
       return FailState::NO_FAILURE;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/media/webrtc/media_stream_capture_indicator.cc chromium-71.0.3578.30/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
index bb2aecf307..af83dec7a3 100644
--- chromium-71.0.3578.30/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
+++ chromium-71.0.3578.30/chrome/browser/media/webrtc/media_stream_capture_indicator.cc
@@ -243,6 +243,7 @@ int& MediaStreamCaptureIndicator::WebContentsDeviceUsage::GetStreamCount(
       NOTREACHED();
       return video_stream_count_;
   }
+  __builtin_abort ();
 }
 
 MediaStreamCaptureIndicator::MediaStreamCaptureIndicator() {}
diff --git chromium-71.0.3578.30/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc chromium-71.0.3578.30/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
index 6a58aadfe4..b8762beb39 100644
--- chromium-71.0.3578.30/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
+++ chromium-71.0.3578.30/chrome/browser/resource_coordinator/tab_lifecycle_unit.cc
@@ -156,6 +156,7 @@ bool IsValidStateChange(LifecycleUnitState from,
       }
     }
   }
+  __builtin_abort ();
 }
 
 StateChangeReason DiscardReasonToStateChangeReason(
@@ -167,6 +168,8 @@ StateChangeReason DiscardReasonToStateChangeReason(
       return StateChangeReason::BROWSER_INITIATED;
     case LifecycleUnitDiscardReason::URGENT:
       return StateChangeReason::SYSTEM_MEMORY_PRESSURE;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/signin/chrome_signin_client.cc chromium-71.0.3578.30/chrome/browser/signin/chrome_signin_client.cc
index e20f0ecc52..bdbcf6a4cc 100644
--- chromium-71.0.3578.30/chrome/browser/signin/chrome_signin_client.cc
+++ chromium-71.0.3578.30/chrome/browser/signin/chrome_signin_client.cc
@@ -105,6 +105,8 @@ SigninClient::SignoutDecision IsSignoutAllowed(
     case signin_metrics::ProfileSignout::NUM_PROFILE_SIGNOUT_METRICS:
       NOTREACHED();
       return SigninClient::SignoutDecision::DISALLOW_SIGNOUT;
+    default:
+      __builtin_abort ();
   }
 }
 }  // namespace
diff --git chromium-71.0.3578.30/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc chromium-71.0.3578.30/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
index 291909441b..cb0d98304e 100644
--- chromium-71.0.3578.30/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/autofill/save_card_bubble_controller_impl.cc
@@ -222,6 +222,8 @@ base::string16 SaveCardBubbleControllerImpl::GetWindowTitle() const {
     case BubbleType::INACTIVE:
       NOTREACHED();
       return base::string16();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/ui/views/autofill/autofill_popup_view_native_views.cc chromium-71.0.3578.30/chrome/browser/ui/views/autofill/autofill_popup_view_native_views.cc
index 8c2109597f..674cea63d6 100644
--- chromium-71.0.3578.30/chrome/browser/ui/views/autofill/autofill_popup_view_native_views.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/views/autofill/autofill_popup_view_native_views.cc
@@ -104,6 +104,8 @@ PopupItemLayoutType GetLayoutType(int frontend_id) {
         default:
           return PopupItemLayoutType::kTrailingIcon;
       }
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -567,6 +569,8 @@ bool AutofillPopupSuggestionView::ShouldUseCustomFontWeightForPrimaryInfo(
     case ForcedFontWeight::kBold:
       *font_weight = gfx::Font::Weight::BOLD;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc chromium-71.0.3578.30/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
index a2598338aa..ce86b5eef5 100644
--- chromium-71.0.3578.30/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/views/autofill/local_card_migration_dialog_view.cc
@@ -139,6 +139,7 @@ bool LocalCardMigrationDialogView::Accept() {
     case LocalCardMigrationDialogState::kActionRequired:
       return true;
   }
+  __builtin_abort ();
 }
 
 bool LocalCardMigrationDialogView::Cancel() {
@@ -151,6 +152,7 @@ bool LocalCardMigrationDialogView::Cancel() {
       controller_->OnViewCardsButtonClicked();
       return true;
   }
+  __builtin_abort ();
 }
 
 void LocalCardMigrationDialogView::OnWidgetClosing(views::Widget* widget) {
@@ -286,6 +288,8 @@ base::string16 LocalCardMigrationDialogView::GetDialogTitle() const {
     case LocalCardMigrationDialogState::kActionRequired:
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_TITLE_FIX);
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -302,6 +306,8 @@ base::string16 LocalCardMigrationDialogView::GetDialogInstruction() const {
     case LocalCardMigrationDialogState::kActionRequired:
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_MESSAGE_FIX);
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -318,6 +324,8 @@ base::string16 LocalCardMigrationDialogView::GetOkButtonLabel() const {
     case LocalCardMigrationDialogState::kActionRequired:
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_BUTTON_LABEL_DONE);
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -331,6 +339,7 @@ base::string16 LocalCardMigrationDialogView::GetCancelButtonLabel() const {
       return l10n_util::GetStringUTF16(
           IDS_AUTOFILL_LOCAL_CARD_MIGRATION_DIALOG_BUTTON_LABEL_VIEW_CARDS);
   }
+  __builtin_abort ();
 }
 
 LocalCardMigrationDialog* CreateLocalCardMigrationDialogView(
diff --git chromium-71.0.3578.30/chrome/browser/ui/views/material_refresh_layout_provider.cc chromium-71.0.3578.30/chrome/browser/ui/views/material_refresh_layout_provider.cc
index 127c5e497f..c5751541da 100644
--- chromium-71.0.3578.30/chrome/browser/ui/views/material_refresh_layout_provider.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/views/material_refresh_layout_provider.cc
@@ -43,6 +43,8 @@ int MaterialRefreshLayoutProvider::GetCornerRadiusMetric(
       return 8;
     case views::EMPHASIS_MAXIMUM:
       return std::min(size.width(), size.height()) / 2;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -60,6 +62,8 @@ int MaterialRefreshLayoutProvider::GetShadowElevationMetric(
       return 3;
     case views::EMPHASIS_MAXIMUM:
       return 16;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/chrome/browser/ui/webui/discards/discards_ui.cc chromium-71.0.3578.30/chrome/browser/ui/webui/discards/discards_ui.cc
index 8717e81f82..0195fcb020 100644
--- chromium-71.0.3578.30/chrome/browser/ui/webui/discards/discards_ui.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/webui/discards/discards_ui.cc
@@ -56,6 +56,8 @@ mojom::LifecycleUnitVisibility GetLifecycleUnitVisibility(
       return mojom::LifecycleUnitVisibility::OCCLUDED;
     case content::Visibility::VISIBLE:
       return mojom::LifecycleUnitVisibility::VISIBLE;
+    default:
+      __builtin_abort ();
   }
 #if defined(COMPILER_MSVC)
   NOTREACHED();
diff --git chromium-71.0.3578.30/chrome/browser/ui/webui/media/media_engagement_ui.cc chromium-71.0.3578.30/chrome/browser/ui/webui/media/media_engagement_ui.cc
index 4a8a58c2d0..eec51ca331 100644
--- chromium-71.0.3578.30/chrome/browser/ui/webui/media/media_engagement_ui.cc
+++ chromium-71.0.3578.30/chrome/browser/ui/webui/media/media_engagement_ui.cc
@@ -100,6 +100,7 @@ class MediaEngagementScoreDetailsProviderImpl
       case content::AutoplayPolicy::kDocumentUserActivationRequired:
         return "document-user-activation-required";
     }
+    __builtin_abort ();
   }
 
   const std::string GetPreloadVersion() {
diff --git chromium-71.0.3578.30/components/autofill/core/browser/autofill_external_delegate.cc chromium-71.0.3578.30/components/autofill/core/browser/autofill_external_delegate.cc
index 33184ca396..5f70a8c80c 100644
--- chromium-71.0.3578.30/components/autofill/core/browser/autofill_external_delegate.cc
+++ chromium-71.0.3578.30/components/autofill/core/browser/autofill_external_delegate.cc
@@ -444,6 +444,8 @@ base::string16 AutofillExternalDelegate::GetSettingsSuggestionValue()
     case PopupType::kPasswords:
       NOTREACHED();
       return base::string16();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/autofill/core/browser/autofill_manager.cc chromium-71.0.3578.30/components/autofill/core/browser/autofill_manager.cc
index f2506032ef..6858e16188 100644
--- chromium-71.0.3578.30/components/autofill/core/browser/autofill_manager.cc
+++ chromium-71.0.3578.30/components/autofill/core/browser/autofill_manager.cc
@@ -257,6 +257,7 @@ PopupType AutofillManager::GetPopupType(const FormData& form,
 
     default:
       NOTREACHED();
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
index dfbede05b0..9645c9c451 100644
--- chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
+++ chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
@@ -28,6 +28,8 @@ sync_pb::WalletMaskedCreditCard::WalletCardStatus LocalToServerStatus(
       return sync_pb::WalletMaskedCreditCard::VALID;
     case CreditCard::EXPIRED:
       return sync_pb::WalletMaskedCreditCard::EXPIRED;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -38,6 +40,8 @@ CreditCard::ServerStatus ServerToLocalStatus(
       return CreditCard::OK;
     case sync_pb::WalletMaskedCreditCard::EXPIRED:
       return CreditCard::EXPIRED;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -82,6 +86,8 @@ const char* CardNetworkFromWalletCardType(
     case sync_pb::WalletMaskedCreditCard::SWITCH:
     case sync_pb::WalletMaskedCreditCard::UNKNOWN:
       return kGenericCard;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -95,6 +101,7 @@ sync_pb::WalletMaskedCreditCard::WalletCardClass WalletCardClassFromCardType(
     case CreditCard::CARD_TYPE_PREPAID:
       return sync_pb::WalletMaskedCreditCard::PREPAID;
     case CreditCard::CARD_TYPE_UNKNOWN:
+    default:
       return sync_pb::WalletMaskedCreditCard::UNKNOWN_CARD_CLASS;
   }
 }
@@ -110,6 +117,8 @@ CreditCard::CardType CardTypeFromWalletCardClass(
       return CreditCard::CARD_TYPE_PREPAID;
     case sync_pb::WalletMaskedCreditCard::UNKNOWN_CARD_CLASS:
       return CreditCard::CARD_TYPE_UNKNOWN;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -162,6 +171,8 @@ std::string GetClientTagForSpecificsId(
     case AutofillWalletSpecifics::UNKNOWN:
       NOTREACHED();
       return "";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_wallet_metadata_sync_bridge.cc chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_wallet_metadata_sync_bridge.cc
index c04756fd18..1618dd8f28 100644
--- chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_wallet_metadata_sync_bridge.cc
+++ chromium-71.0.3578.30/components/autofill/core/browser/webdata/autofill_wallet_metadata_sync_bridge.cc
@@ -43,6 +43,8 @@ std::string GetClientTagForSpecificsId(WalletMetadataSpecifics::Type type,
     case WalletMetadataSpecifics::UNKNOWN:
       NOTREACHED();
       return "";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/cast_channel/cast_message_util.cc chromium-71.0.3578.30/components/cast_channel/cast_message_util.cc
index 0dda189424..d54c29d1f6 100644
--- chromium-71.0.3578.30/components/cast_channel/cast_message_util.cc
+++ chromium-71.0.3578.30/components/cast_channel/cast_message_util.cc
@@ -362,6 +362,8 @@ const char* GetAppAvailabilityResultToString(GetAppAvailabilityResult result) {
       return "unavailable";
     case GetAppAvailabilityResult::kUnknown:
       return "unknown";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc chromium-71.0.3578.30/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
index 2deca8cdbc..5c659d960b 100644
--- chromium-71.0.3578.30/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
+++ chromium-71.0.3578.30/components/data_reduction_proxy/core/browser/data_reduction_proxy_util.cc
@@ -337,6 +337,8 @@ PageloadMetrics_ConnectionType ProtoConnectionTypeFromConnectionType(
       return PageloadMetrics_ConnectionType_CONNECTION_NONE;
     case net::NetworkChangeNotifier::CONNECTION_BLUETOOTH:
       return PageloadMetrics_ConnectionType_CONNECTION_BLUETOOTH;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -351,6 +353,8 @@ RequestInfo_Protocol ProtoRequestInfoProtocolFromRequestInfoProtocol(
       return RequestInfo_Protocol_QUIC;
     case DataReductionProxyData::RequestInfo::Protocol::UNKNOWN:
       return RequestInfo_Protocol_UNKNOWN;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/password_manager/core/browser/browser_save_password_progress_logger.cc chromium-71.0.3578.30/components/password_manager/core/browser/browser_save_password_progress_logger.cc
index 5524cc6c73..6f1895343d 100644
--- chromium-71.0.3578.30/components/password_manager/core/browser/browser_save_password_progress_logger.cc
+++ chromium-71.0.3578.30/components/password_manager/core/browser/browser_save_password_progress_logger.cc
@@ -72,6 +72,8 @@ std::string VoteTypeToString(
       return "HTML classifier";
     case AutofillUploadContents::Field::FIRST_USE:
       return "First use";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/policy/core/common/cloud/cloud_policy_util.cc chromium-71.0.3578.30/components/policy/core/common/cloud/cloud_policy_util.cc
index 897d87f3fb..57cc3b083c 100644
--- chromium-71.0.3578.30/components/policy/core/common/cloud/cloud_policy_util.cc
+++ chromium-71.0.3578.30/components/policy/core/common/cloud/cloud_policy_util.cc
@@ -170,6 +170,8 @@ em::Channel ConvertToProtoChannel(version_info::Channel channel) {
       return em::CHANNEL_BETA;
     case version_info::Channel::STABLE:
       return em::CHANNEL_STABLE;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/policy/core/common/policy_proto_decoders.cc chromium-71.0.3578.30/components/policy/core/common/policy_proto_decoders.cc
index 5f13eddca6..83d38182b8 100644
--- chromium-71.0.3578.30/components/policy/core/common/policy_proto_decoders.cc
+++ chromium-71.0.3578.30/components/policy/core/common/policy_proto_decoders.cc
@@ -42,6 +42,8 @@ bool GetPolicyLevel(const AnyPolicyProto& policy_proto, PolicyLevel* level) {
       return true;
     case em::PolicyOptions::UNSET:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/previews/content/previews_hints.cc chromium-71.0.3578.30/components/previews/content/previews_hints.cc
index 41c353f249..9a4b54b8c9 100644
--- chromium-71.0.3578.30/components/previews/content/previews_hints.cc
+++ chromium-71.0.3578.30/components/previews/content/previews_hints.cc
@@ -123,6 +123,8 @@ ConvertProtoOptimizationTypeToPreviewsOptimizationType(
       return PreviewsType::RESOURCE_LOADING_HINTS;
     case optimization_guide::proto::LITE_PAGE_REDIRECT:
       return PreviewsType::LITE_PAGE_REDIRECT;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/previews/core/previews_black_list.cc chromium-71.0.3578.30/components/previews/core/previews_black_list.cc
index d718e8412e..6213d2b305 100644
--- chromium-71.0.3578.30/components/previews/core/previews_black_list.cc
+++ chromium-71.0.3578.30/components/previews/core/previews_black_list.cc
@@ -33,6 +33,8 @@ PreviewsEligibilityReason BlacklistReasonToPreviewsReason(
       return PreviewsEligibilityReason::ALLOWED;
     case blacklist::BlacklistReason::kAllowed:
       return PreviewsEligibilityReason::ALLOWED;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/signin/core/browser/about_signin_internals.cc chromium-71.0.3578.30/components/signin/core/browser/about_signin_internals.cc
index b5ceb2792f..47fab318f7 100644
--- chromium-71.0.3578.30/components/signin/core/browser/about_signin_internals.cc
+++ chromium-71.0.3578.30/components/signin/core/browser/about_signin_internals.cc
@@ -539,6 +539,8 @@ std::string AboutSigninInternals::RefreshTokenEvent::GetTypeAsString() const {
       return "Revoked";
     case AboutSigninInternals::RefreshTokenEventType::kAllTokensLoaded:
       return "Loaded";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/signin/core/browser/profile_oauth2_token_service.cc chromium-71.0.3578.30/components/signin/core/browser/profile_oauth2_token_service.cc
index e840a3bf96..db21daf276 100644
--- chromium-71.0.3578.30/components/signin/core/browser/profile_oauth2_token_service.cc
+++ chromium-71.0.3578.30/components/signin/core/browser/profile_oauth2_token_service.cc
@@ -121,6 +121,8 @@ bool ProfileOAuth2TokenService::HasLoadCredentialsFinishedWithNoErrors() {
         LOAD_CREDENTIALS_FINISHED_WITH_NO_TOKEN_FOR_PRIMARY_ACCOUNT:
       // Load credentials finished with success.
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/components/viz/service/display/skia_renderer.cc chromium-71.0.3578.30/components/viz/service/display/skia_renderer.cc
index 58fb3ca97d..68f9dbb99d 100644
--- chromium-71.0.3578.30/components/viz/service/display/skia_renderer.cc
+++ chromium-71.0.3578.30/components/viz/service/display/skia_renderer.cc
@@ -1055,6 +1055,8 @@ GrContext* SkiaRenderer::GetGrContext() {
 #endif
     case DrawMode::GL:
       return context_provider_->GrContext();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/content/browser/cache_storage/cache_storage_quota_client.cc chromium-71.0.3578.30/content/browser/cache_storage/cache_storage_quota_client.cc
index 50086c0cd9..8d578a00a5 100644
--- chromium-71.0.3578.30/content/browser/cache_storage/cache_storage_quota_client.cc
+++ chromium-71.0.3578.30/content/browser/cache_storage/cache_storage_quota_client.cc
@@ -98,6 +98,8 @@ storage::QuotaClient::ID CacheStorageQuotaClient::GetIDFromOwner(
       return kServiceWorkerCache;
     case CacheStorageOwner::kBackgroundFetch:
       return kBackgroundFetch;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/content/browser/frame_host/frame_tree_node.cc chromium-71.0.3578.30/content/browser/frame_host/frame_tree_node.cc
index 24f5fbb94a..f8a1cebdf1 100644
--- chromium-71.0.3578.30/content/browser/frame_host/frame_tree_node.cc
+++ chromium-71.0.3578.30/content/browser/frame_host/frame_tree_node.cc
@@ -566,6 +566,7 @@ bool FrameTreeNode::UpdateUserActivationState(
       return NotifyUserActivation();
   }
   NOTREACHED() << "Invalid update_type.";
+  __builtin_abort ();
 }
 
 void FrameTreeNode::OnSetHasReceivedUserGestureBeforeNavigation(bool value) {
diff --git chromium-71.0.3578.30/content/browser/media/audio_output_stream_broker.cc chromium-71.0.3578.30/content/browser/media/audio_output_stream_broker.cc
index 2ee519d96d..5f6c053ec3 100644
--- chromium-71.0.3578.30/content/browser/media/audio_output_stream_broker.cc
+++ chromium-71.0.3578.30/content/browser/media/audio_output_stream_broker.cc
@@ -53,6 +53,8 @@ StreamBrokerDisconnectReason GetDisconnectReason(DisconnectReason reason,
                  : StreamBrokerDisconnectReason::kDocumentDestroyed;
     case DisconnectReason::kDefault:
       return StreamBrokerDisconnectReason::kDefault;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/content/renderer/input/input_event_prediction.cc chromium-71.0.3578.30/content/renderer/input/input_event_prediction.cc
index 5b0579d360..f543562027 100644
--- chromium-71.0.3578.30/content/renderer/input/input_event_prediction.cc
+++ chromium-71.0.3578.30/content/renderer/input/input_event_prediction.cc
@@ -76,6 +76,8 @@ std::unique_ptr<ui::InputPredictor> InputEventPrediction::CreatePredictor()
       return std::make_unique<ui::LeastSquaresPredictor>();
     case PredictorType::kKalman:
       return std::make_unique<ui::KalmanPredictor>();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/content/renderer/media/stream/media_stream_audio_processor_options.cc chromium-71.0.3578.30/content/renderer/media/stream/media_stream_audio_processor_options.cc
index e6b525b05b..1682f3f097 100644
--- chromium-71.0.3578.30/content/renderer/media/stream/media_stream_audio_processor_options.cc
+++ chromium-71.0.3578.30/content/renderer/media/stream/media_stream_audio_processor_options.cc
@@ -70,6 +70,8 @@ AudioProcessingProperties::ToAudioProcessingSettings() const {
         return media::EchoCancellationType::kAec3;
       case EchoCancellationType::kEchoCancellationSystem:
         return media::EchoCancellationType::kSystemAec;
+      default:
+        __builtin_abort ();
     }
   };
 
diff --git chromium-71.0.3578.30/content/renderer/media/stream/media_stream_constraints_util_video_device.cc chromium-71.0.3578.30/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
index 8be1cee604..40d8a16b29 100644
--- chromium-71.0.3578.30/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
+++ chromium-71.0.3578.30/content/renderer/media/stream/media_stream_constraints_util_video_device.cc
@@ -739,6 +739,8 @@ blink::WebMediaStreamTrack::DisplayCaptureSurfaceType ToWebDisplaySurface(
           kApplication;
     case media::mojom::DisplayCaptureSurfaceType::BROWSER:
       return blink::WebMediaStreamTrack::DisplayCaptureSurfaceType::kBrowser;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -751,6 +753,8 @@ blink::WebMediaStreamTrack::CursorCaptureType ToWebCursorCaptureType(
       return blink::WebMediaStreamTrack::CursorCaptureType::kAlways;
     case media::mojom::CursorCaptureType::MOTION:
       return blink::WebMediaStreamTrack::CursorCaptureType::kMotion;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/content/renderer/media/webrtc/peer_connection_tracker.cc chromium-71.0.3578.30/content/renderer/media/webrtc/peer_connection_tracker.cc
index 4af9ed31a9..09ecb35361 100644
--- chromium-71.0.3578.30/content/renderer/media/webrtc/peer_connection_tracker.cc
+++ chromium-71.0.3578.30/content/renderer/media/webrtc/peer_connection_tracker.cc
@@ -118,6 +118,8 @@ static const char* SerializeDirection(
       return "'recvonly'";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "'inactive'";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/device/bluetooth/bluetooth_remote_gatt_characteristic.cc chromium-71.0.3578.30/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
index c81a498511..3c9c41c952 100644
--- chromium-71.0.3578.30/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
+++ chromium-71.0.3578.30/device/bluetooth/bluetooth_remote_gatt_characteristic.cc
@@ -396,6 +396,8 @@ bool BluetoothRemoteGattCharacteristic::IsNotificationTypeSupported(
       return hasNotify;
     case NotificationType::kIndication:
       return hasIndicate;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/extensions/browser/url_loader_factory_manager.cc chromium-71.0.3578.30/extensions/browser/url_loader_factory_manager.cc
index eb8bd9c495..7bd5e8d83a 100644
--- chromium-71.0.3578.30/extensions/browser/url_loader_factory_manager.cc
+++ chromium-71.0.3578.30/extensions/browser/url_loader_factory_manager.cc
@@ -64,6 +64,8 @@ bool IsSpecialURLLoaderFactoryRequired(const Extension& extension,
              DoExtensionPermissionsCoverCorsOrCorbRelatedOrigins(extension);
     case FactoryUser::kExtensionProcess:
       return DoExtensionPermissionsCoverCorsOrCorbRelatedOrigins(extension);
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/gin/v8_isolate_memory_dump_provider.cc chromium-71.0.3578.30/gin/v8_isolate_memory_dump_provider.cc
index fe14263836..2df77c8454 100644
--- chromium-71.0.3578.30/gin/v8_isolate_memory_dump_provider.cc
+++ chromium-71.0.3578.30/gin/v8_isolate_memory_dump_provider.cc
@@ -131,6 +131,8 @@ std::string IsolateTypeString(IsolateHolder::IsolateType isolate_type) {
       return "test";
     case IsolateHolder::IsolateType::kUtility:
       return "utility";
+    default:
+      __builtin_abort ();
   }
   LOG(FATAL) << "Unreachable code";
 }
@@ -147,6 +149,8 @@ bool CanHaveMultipleIsolates(IsolateHolder::IsolateType isolate_type) {
     case IsolateHolder::IsolateType::kUtility:
       // PDFium and ProxyResolver create one isolate per process.
       return false;
+    default:
+      __builtin_abort ();
   }
   LOG(FATAL) << "Unreachable code";
 }
diff --git chromium-71.0.3578.30/gpu/config/gpu_info.cc chromium-71.0.3578.30/gpu/config/gpu_info.cc
index 9c44106fe2..4de7ae7008 100644
--- chromium-71.0.3578.30/gpu/config/gpu_info.cc
+++ chromium-71.0.3578.30/gpu/config/gpu_info.cc
@@ -83,6 +83,8 @@ const char* OverlayFormatToString(OverlayFormat format) {
       return "YUY2";
     case OverlayFormat::kNV12:
       return "NV12";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/audio/audio_processing.cc chromium-71.0.3578.30/media/audio/audio_processing.cc
index b9af6691c4..987f7536c7 100644
--- chromium-71.0.3578.30/media/audio/audio_processing.cc
+++ chromium-71.0.3578.30/media/audio/audio_processing.cc
@@ -19,6 +19,8 @@ std::string AudioProcessingSettings::ToString() const {
         return "experimental";
       case AutomaticGainControlType::kHybridExperimental:
         return "hybrid experimental";
+      default:
+        __builtin_abort ();
     }
   };
 
@@ -32,6 +34,8 @@ std::string AudioProcessingSettings::ToString() const {
         return "aec3";
       case EchoCancellationType::kSystemAec:
         return "system aec";
+      default:
+        __builtin_abort ();
     }
   };
 
@@ -43,6 +47,8 @@ std::string AudioProcessingSettings::ToString() const {
         return "default";
       case NoiseSuppressionType::kExperimental:
         return "experimental";
+      default:
+        __builtin_abort ();
     }
   };
 
diff --git chromium-71.0.3578.30/media/base/decode_status.cc chromium-71.0.3578.30/media/base/decode_status.cc
index 110696fd94..a8c7533c7f 100644
--- chromium-71.0.3578.30/media/base/decode_status.cc
+++ chromium-71.0.3578.30/media/base/decode_status.cc
@@ -18,6 +18,8 @@ const char* GetDecodeStatusString(DecodeStatus status) {
       return "DecodeStatus::ABORTED";
     case DecodeStatus::DECODE_ERROR:
       return "DecodeStatus::DECODE_ERROR";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/base/decryptor.cc chromium-71.0.3578.30/media/base/decryptor.cc
index f37fda5ad4..ffe839cef6 100644
--- chromium-71.0.3578.30/media/base/decryptor.cc
+++ chromium-71.0.3578.30/media/base/decryptor.cc
@@ -17,6 +17,8 @@ const char* Decryptor::GetStatusName(Status status) {
       return "need_more_data";
     case kError:
       return "error";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/base/demuxer_stream.cc chromium-71.0.3578.30/media/base/demuxer_stream.cc
index 9fad473f2d..6b2152e92d 100644
--- chromium-71.0.3578.30/media/base/demuxer_stream.cc
+++ chromium-71.0.3578.30/media/base/demuxer_stream.cc
@@ -17,6 +17,8 @@ const char* DemuxerStream::GetTypeName(Type type) {
       return "text";
     case DemuxerStream::UNKNOWN:
       return "unknown";
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -31,6 +33,8 @@ const char* DemuxerStream::GetStatusName(Status status) {
       return "config_changed";
     case DemuxerStream::kError:
       return "error";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/blink/watch_time_reporter.cc chromium-71.0.3578.30/media/blink/watch_time_reporter.cc
index c4c15d154c..4310bb8143 100644
--- chromium-71.0.3578.30/media/blink/watch_time_reporter.cc
+++ chromium-71.0.3578.30/media/blink/watch_time_reporter.cc
@@ -571,6 +571,8 @@ WatchTimeKey WatchTimeReporter::GetDisplayTypeKey(DisplayType display_type) {
       return DISPLAY_TYPE_KEY(DisplayFullscreen);
     case DisplayType::kPictureInPicture:
       return DISPLAY_TYPE_KEY(DisplayPictureInPicture);
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/blink/webmediaplayer_impl.cc chromium-71.0.3578.30/media/blink/webmediaplayer_impl.cc
index 70d506014a..ad202427ae 100644
--- chromium-71.0.3578.30/media/blink/webmediaplayer_impl.cc
+++ chromium-71.0.3578.30/media/blink/webmediaplayer_impl.cc
@@ -196,6 +196,8 @@ EncryptionMode DetermineEncryptionMode(
       return EncryptionMode::kCenc;
     case EncryptionScheme::CIPHER_MODE_AES_CBC:
       return EncryptionMode::kCbcs;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/capture/video/video_capture_device.cc chromium-71.0.3578.30/media/capture/video/video_capture_device.cc
index d74830a059..f5303f5059 100644
--- chromium-71.0.3578.30/media/capture/video/video_capture_device.cc
+++ chromium-71.0.3578.30/media/capture/video/video_capture_device.cc
@@ -83,6 +83,8 @@ VideoCaptureFrameDropReason ConvertReservationFailureToFrameDropReason(
       return VideoCaptureFrameDropReason::kBufferPoolMaxBufferCountExceeded;
     case VideoCaptureDevice::Client::ReserveResult::kAllocationFailed:
       return VideoCaptureFrameDropReason::kBufferPoolBufferAllocationFailed;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/capture/video/video_capture_device_descriptor.cc chromium-71.0.3578.30/media/capture/video/video_capture_device_descriptor.cc
index be6c87ec4c..fd4be4889d 100644
--- chromium-71.0.3578.30/media/capture/video/video_capture_device_descriptor.cc
+++ chromium-71.0.3578.30/media/capture/video/video_capture_device_descriptor.cc
@@ -94,6 +94,8 @@ const char* VideoCaptureDeviceDescriptor::GetCaptureApiTypeString() const {
       return "Virtual Device";
     case VideoCaptureApi::UNKNOWN:
       return "Unknown";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/cdm/cdm_helpers.cc chromium-71.0.3578.30/media/cdm/cdm_helpers.cc
index eb98759559..c7b37a9297 100644
--- chromium-71.0.3578.30/media/cdm/cdm_helpers.cc
+++ chromium-71.0.3578.30/media/cdm/cdm_helpers.cc
@@ -25,6 +25,8 @@ gfx::ColorSpace::RangeID ToGfxColorSpaceRange(cdm::ColorRange range) {
       return gfx::ColorSpace::RangeID::FULL;
     case cdm::ColorRange::kDerived:
       return gfx::ColorSpace::RangeID::DERIVED;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/filters/decoder_stream.cc chromium-71.0.3578.30/media/filters/decoder_stream.cc
index 062ad30a4f..106228a77c 100644
--- chromium-71.0.3578.30/media/filters/decoder_stream.cc
+++ chromium-71.0.3578.30/media/filters/decoder_stream.cc
@@ -89,6 +89,8 @@ const char* GetStatusString(typename DecoderStream<StreamType>::Status status) {
       return "demuxer_read_aborted";
     case DecoderStream<StreamType>::DECODE_ERROR:
       return "decode_error";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/media/gpu/vaapi/vaapi_wrapper.cc chromium-71.0.3578.30/media/gpu/vaapi/vaapi_wrapper.cc
index e575c0329a..75fba8d485 100644
--- chromium-71.0.3578.30/media/gpu/vaapi/vaapi_wrapper.cc
+++ chromium-71.0.3578.30/media/gpu/vaapi/vaapi_wrapper.cc
@@ -401,6 +401,8 @@ static VAEntrypoint GetVaEntryPoint(VaapiWrapper::CodecMode mode,
     case VaapiWrapper::kCodecModeMax:
       NOTREACHED();
       return VAEntrypointVLD;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/net/nqe/network_quality_estimator.cc chromium-71.0.3578.30/net/nqe/network_quality_estimator.cc
index 71d1565c94..aedc557a18 100644
--- chromium-71.0.3578.30/net/nqe/network_quality_estimator.cc
+++ chromium-71.0.3578.30/net/nqe/network_quality_estimator.cc
@@ -1390,6 +1390,8 @@ base::TimeDelta NetworkQualityEstimator::GetRTTEstimateInternal(
     case nqe::internal::OBSERVATION_CATEGORY_COUNT:
       NOTREACHED();
       return base::TimeDelta();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/net/third_party/quic/core/qpack/qpack_decoder.cc chromium-71.0.3578.30/net/third_party/quic/core/qpack/qpack_decoder.cc
index 13d1fc1509..dcfa6eebcc 100644
--- chromium-71.0.3578.30/net/third_party/quic/core/qpack/qpack_decoder.cc
+++ chromium-71.0.3578.30/net/third_party/quic/core/qpack/qpack_decoder.cc
@@ -168,6 +168,8 @@ size_t QpackDecoder::ProgressiveDecoder::DoStart(QuicStringPiece data) {
     case http2::DecodeStatus::kDecodeError:
       OnError("Encoded integer too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -189,6 +191,8 @@ size_t QpackDecoder::ProgressiveDecoder::DoVarintResume(QuicStringPiece data) {
     case http2::DecodeStatus::kDecodeError:
       OnError("Encoded integer too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -281,6 +285,8 @@ size_t QpackDecoder::ProgressiveDecoder::DoValueLengthStart(
     case http2::DecodeStatus::kDecodeError:
       OnError("ValueLen too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -303,6 +309,8 @@ size_t QpackDecoder::ProgressiveDecoder::DoValueLengthResume(
     case http2::DecodeStatus::kDecodeError:
       OnError("ValueLen too large.");
       return bytes_consumed;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/services/audio/public/cpp/audio_system_to_service_adapter.cc chromium-71.0.3578.30/services/audio/public/cpp/audio_system_to_service_adapter.cc
index 91967b5060..0e0ea0409d 100644
--- chromium-71.0.3578.30/services/audio/public/cpp/audio_system_to_service_adapter.cc
+++ chromium-71.0.3578.30/services/audio/public/cpp/audio_system_to_service_adapter.cc
@@ -66,6 +66,8 @@ const char* GetTraceEvent(Action action) {
       return "AudioSystemToServiceAdapter::GetAssociatedOutputDeviceID";
     case kGetInputDeviceInfo:
       return "AudioSystemToServiceAdapter::GetInputDeviceInfo";
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
diff --git chromium-71.0.3578.30/services/network/cross_origin_read_blocking.cc chromium-71.0.3578.30/services/network/cross_origin_read_blocking.cc
index 02b7e5f6ec..f6430fc81b 100644
--- chromium-71.0.3578.30/services/network/cross_origin_read_blocking.cc
+++ chromium-71.0.3578.30/services/network/cross_origin_read_blocking.cc
@@ -782,6 +782,8 @@ bool CrossOriginReadBlocking::ResponseAnalyzer::ShouldAllow() const {
       return sniffers_.empty() && !found_blockable_content_;
     case kBlock:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -793,6 +795,8 @@ bool CrossOriginReadBlocking::ResponseAnalyzer::ShouldBlock() const {
       return sniffers_.empty() && found_blockable_content_;
     case kBlock:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/services/network/public/cpp/cors/cors.cc chromium-71.0.3578.30/services/network/public/cpp/cors/cors.cc
index 9e99fa3fb4..46e326341d 100644
--- chromium-71.0.3578.30/services/network/public/cpp/cors/cors.cc
+++ chromium-71.0.3578.30/services/network/public/cpp/cors/cors.cc
@@ -537,6 +537,8 @@ bool IsCORSSameOriginResponseType(mojom::FetchResponseType type) {
     case mojom::FetchResponseType::kOpaque:
     case mojom::FetchResponseType::kOpaqueRedirect:
       return false;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -550,6 +552,8 @@ bool IsCORSCrossOriginResponseType(mojom::FetchResponseType type) {
     case mojom::FetchResponseType::kOpaque:
     case mojom::FetchResponseType::kOpaqueRedirect:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -567,6 +571,8 @@ bool CalculateCredentialsFlag(mojom::FetchCredentialsMode credentials_mode,
       return response_tainting == network::mojom::FetchResponseType::kBasic;
     case network::mojom::FetchCredentialsMode::kInclude:
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/services/network/public/cpp/host_resolver_mojom_traits.cc chromium-71.0.3578.30/services/network/public/cpp/host_resolver_mojom_traits.cc
index 0364c3e0d8..01fb792e5f 100644
--- chromium-71.0.3578.30/services/network/public/cpp/host_resolver_mojom_traits.cc
+++ chromium-71.0.3578.30/services/network/public/cpp/host_resolver_mojom_traits.cc
@@ -20,6 +20,8 @@ ResolveHostParameters::DnsQueryType EnumTraits<
       return ResolveHostParameters::DnsQueryType::A;
     case net::HostResolver::DnsQueryType::AAAA:
       return ResolveHostParameters::DnsQueryType::AAAA;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -38,6 +40,8 @@ bool EnumTraits<ResolveHostParameters::DnsQueryType,
     case ResolveHostParameters::DnsQueryType::AAAA:
       *output = net::HostResolver::DnsQueryType::AAAA;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -54,6 +58,8 @@ EnumTraits<ResolveHostParameters::Source, net::HostResolverSource>::ToMojom(
       return ResolveHostParameters::Source::DNS;
     case net::HostResolverSource::MULTICAST_DNS:
       return ResolveHostParameters::Source::MULTICAST_DNS;
+    default:
+      __builtin_abort ();
   }
 }
 
@@ -74,6 +80,8 @@ bool EnumTraits<ResolveHostParameters::Source, net::HostResolverSource>::
     case ResolveHostParameters::Source::MULTICAST_DNS:
       *output = net::HostResolverSource::MULTICAST_DNS;
       return true;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/services/ws/public/cpp/gpu/command_buffer_metrics.cc chromium-71.0.3578.30/services/ws/public/cpp/gpu/command_buffer_metrics.cc
index 2671ff8cb9..97fe06a6ad 100644
--- chromium-71.0.3578.30/services/ws/public/cpp/gpu/command_buffer_metrics.cc
+++ chromium-71.0.3578.30/services/ws/public/cpp/gpu/command_buffer_metrics.cc
@@ -96,6 +96,8 @@ std::string ContextTypeToString(ContextType type) {
       return "ForTesting";
     case ContextType::XR_COMPOSITING:
       return "XRCompositing";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/core/css/style_environment_variables.cc chromium-71.0.3578.30/third_party/blink/renderer/core/css/style_environment_variables.cc
index 4f76fb1eb7..a60335a8d7 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/core/css/style_environment_variables.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/core/css/style_environment_variables.cc
@@ -63,7 +63,7 @@ const AtomicString StyleEnvironmentVariables::GetVariableName(
     case UADefinedVariable::kSafeAreaInsetRight:
       return "safe-area-inset-right";
     default:
-      break;
+      __builtin_abort ();
   }
 
   NOTREACHED();
diff --git chromium-71.0.3578.30/third_party/blink/renderer/core/html/media/media_controls.cc chromium-71.0.3578.30/third_party/blink/renderer/core/html/media/media_controls.cc
index d24af6e56f..a97855ddc9 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/core/html/media/media_controls.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/core/html/media/media_controls.cc
@@ -36,6 +36,8 @@ AtomicString MediaControls::GetSizingCSSClass(
       return kMediaControlsSizingMediumCSSClass;
     case MediaControlsSizingClass::kLarge:
       return kMediaControlsSizingLargeCSSClass;
+    default:
+      __builtin_abort ();
   }
 
   NOTREACHED();
diff --git chromium-71.0.3578.30/third_party/blink/renderer/core/layout/layout_flexible_box.cc chromium-71.0.3578.30/third_party/blink/renderer/core/layout/layout_flexible_box.cc
index 6fdc8c1301..762a8d10d8 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/core/layout/layout_flexible_box.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/core/layout/layout_flexible_box.cc
@@ -592,6 +592,8 @@ LayoutUnit LayoutFlexibleBox::FlowAwareContentInsetBefore() const {
       return ContentLeft();
     case TransformedWritingMode::kRightToLeftWritingMode:
       return ContentInsetRight();
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
@@ -607,6 +609,8 @@ LayoutUnit LayoutFlexibleBox::FlowAwareContentInsetAfter() const {
       return ContentInsetRight();
     case TransformedWritingMode::kRightToLeftWritingMode:
       return ContentLeft();
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
diff --git chromium-71.0.3578.30/third_party/blink/renderer/core/loader/mixed_content_checker.cc chromium-71.0.3578.30/third_party/blink/renderer/core/loader/mixed_content_checker.cc
index 4355ac7084..36116e9f02 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/core/loader/mixed_content_checker.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/core/loader/mixed_content_checker.cc
@@ -245,6 +245,8 @@ bool MixedContentChecker::IsMixedContent(
 
     case HttpsState::kModern:
       return IsInsecureUrl(url);
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
index c61588ea1d..d3811b683d 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_layout_object.cc
@@ -1298,6 +1298,8 @@ ax::mojom::TextPosition AXLayoutObject::GetTextPosition() const {
       return ax::mojom::TextPosition::kSubscript;
     case EVerticalAlign::kSuper:
       return ax::mojom::TextPosition::kSuperscript;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_position.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_position.cc
index dd812d09ee..a0e5cea7c4 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_position.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/accessibility/ax_position.cc
@@ -710,6 +710,8 @@ const AXObject* AXPosition::FindNeighboringUnignoredObject(
       }
       return nullptr;
     }
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
index e425272b6d..640c6b58d2 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/background_fetch/background_fetch_registration.cc
@@ -303,6 +303,8 @@ const String BackgroundFetchRegistration::result() const {
       return "failure";
     case mojom::BackgroundFetchResult::UNSET:
       return "";
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
@@ -323,6 +325,8 @@ const String BackgroundFetchRegistration::failureReason() const {
       return "quota-exceeded";
     case mojom::BackgroundFetchFailureReason::TOTAL_DOWNLOAD_SIZE_EXCEEDED:
       return "total-download-exceeded";
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
index c91e99e779..2d3ad08958 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/cookie_store/cookie_change_event.cc
@@ -53,6 +53,8 @@ String ToCookieListItemSameSite(network::mojom::CookieSameSite same_site) {
       return "lax";
     case network::mojom::CookieSameSite::NO_RESTRICTION:
       return "unrestricted";
+    default:
+      __builtin_abort ();
   }
 
   NOTREACHED();
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
index 91e6757baf..7190134c33 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/peerconnection/rtc_rtp_transceiver.cc
@@ -28,6 +28,8 @@ String TransceiverDirectionToString(
       return "recvonly";
     case webrtc::RtpTransceiverDirection::kInactive:
       return "inactive";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/storage/cached_storage_area.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/storage/cached_storage_area.cc
index afc5524c53..2225081901 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/storage/cached_storage_area.cc
@@ -636,6 +636,8 @@ Vector<uint8_t> CachedStorageArea::StringToUint8Vector(
                   input.length() * sizeof(UChar));
       return result;
     }
+    default:
+      __builtin_abort ();
   }
   NOTREACHED();
 }
diff --git chromium-71.0.3578.30/third_party/blink/renderer/modules/webaudio/audio_param.cc chromium-71.0.3578.30/third_party/blink/renderer/modules/webaudio/audio_param.cc
index 24f4820c89..b4a1ffee7d 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/modules/webaudio/audio_param.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/modules/webaudio/audio_param.cc
@@ -145,7 +145,7 @@ String AudioParamHandler::GetParamName() const {
     case kParamTypeAudioWorklet:
       return custom_param_name_;
     default:
-      NOTREACHED();
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc chromium-71.0.3578.30/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
index 906bf2bf45..72f8f5b4f4 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.cc
@@ -18,7 +18,7 @@ WTF::String OverscrollBehaviorTypeToString(
     case OverscrollBehavior::kOverscrollBehaviorTypeContain:
       return "contain";
     default:
-      NOTREACHED();
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/platform/heap/heap_stats_collector.h chromium-71.0.3578.30/third_party/blink/renderer/platform/heap/heap_stats_collector.h
index 05680bbd4d..0b88198f8a 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/platform/heap/heap_stats_collector.h
+++ chromium-71.0.3578.30/third_party/blink/renderer/platform/heap/heap_stats_collector.h
@@ -94,6 +94,8 @@ class PLATFORM_EXPORT ThreadHeapStatsCollector {
         return "BlinkGC.VisitPersistents";
       case Id::kVisitStackRoots:
         return "BlinkGC.VisitStackRoots";
+      default:
+        __builtin_abort ();
     }
   }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
index a06c71d0f1..9b8fb53ad1 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/common/metrics_helper.cc
@@ -44,6 +44,8 @@ scheduling_metrics::ThreadType ConvertBlinkThreadType(
     case WebThreadType::kCount:
       NOTREACHED();
       return scheduling_metrics::ThreadType::kCount;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
index c713c6f94a..00f4ccb12a 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
+++ chromium-71.0.3578.30/third_party/blink/renderer/platform/scheduler/main_thread/page_scheduler_impl.cc
@@ -676,6 +676,8 @@ PageSchedulerImpl::PageLifecycleStateTracker::
           NOTREACHED();
           return base::nullopt;
       }
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/blink/renderer/platform/wtf/casting.h chromium-71.0.3578.30/third_party/blink/renderer/platform/wtf/casting.h
index ee2873f573..09951be05f 100644
--- chromium-71.0.3578.30/third_party/blink/renderer/platform/wtf/casting.h
+++ chromium-71.0.3578.30/third_party/blink/renderer/platform/wtf/casting.h
@@ -57,6 +57,7 @@ struct DowncastTraits {
   template <typename U>
   static bool AllowFrom(const U&) {
     static_assert(sizeof(U) == 0, "no downcast traits specialization for T");
+    return false;
   }
 };
 
diff --git chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
index c0a736a4ed..6b0625fd86 100644
--- chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
+++ chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/temporal_layers_checker.cc
@@ -26,6 +26,8 @@ TemporalLayersChecker::CreateTemporalLayersChecker(TemporalLayersType type,
     case TemporalLayersType::kBitrateDynamic:
       // Conference mode temporal layering for screen content in base stream.
       return absl::make_unique<TemporalLayersChecker>(num_temporal_layers);
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_temporal_layers.cc chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_temporal_layers.cc
index 49584414a9..0a302fca9c 100644
--- chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_temporal_layers.cc
+++ chromium-71.0.3578.30/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_temporal_layers.cc
@@ -36,6 +36,8 @@ std::unique_ptr<TemporalLayers> TemporalLayers::CreateTemporalLayers(
       // Conference mode temporal layering for screen content in base stream.
       return absl::make_unique<ScreenshareLayers>(num_temporal_layers,
                                                   Clock::GetRealTimeClock());
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/ui/accessibility/platform/ax_platform_node_auralinux.cc chromium-71.0.3578.30/ui/accessibility/platform/ax_platform_node_auralinux.cc
index c61d9ea4e1..4069c24a34 100644
--- chromium-71.0.3578.30/ui/accessibility/platform/ax_platform_node_auralinux.cc
+++ chromium-71.0.3578.30/ui/accessibility/platform/ax_platform_node_auralinux.cc
@@ -1584,6 +1584,8 @@ AtkRole AXPlatformNodeAuraLinux::GetAtkRole() {
     case ax::mojom::Role::kPresentational:
     case ax::mojom::Role::kUnknown:
       return ATK_ROLE_REDUNDANT_OBJECT;
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/ui/aura/mus/in_flight_change.cc chromium-71.0.3578.30/ui/aura/mus/in_flight_change.cc
index fe5ed3153b..546dcc2ee6 100644
--- chromium-71.0.3578.30/ui/aura/mus/in_flight_change.cc
+++ chromium-71.0.3578.30/ui/aura/mus/in_flight_change.cc
@@ -57,6 +57,8 @@ std::string ChangeTypeToString(ChangeType change_type) {
       return "TRANSFORM";
     case ChangeType::VISIBLE:
       return "VISIBLE";
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/ui/base/mojo/clipboard_struct_traits.h chromium-71.0.3578.30/ui/base/mojo/clipboard_struct_traits.h
index 518cb04413..aa222323a3 100644
--- chromium-71.0.3578.30/ui/base/mojo/clipboard_struct_traits.h
+++ chromium-71.0.3578.30/ui/base/mojo/clipboard_struct_traits.h
@@ -20,6 +20,8 @@ struct EnumTraits<ui::mojom::ClipboardType, ui::ClipboardType> {
         return ui::mojom::ClipboardType::SELECTION;
       case ui::CLIPBOARD_TYPE_DRAG:
         return ui::mojom::ClipboardType::DRAG;
+      default:
+        __builtin_abort ();
     }
   }
 
@@ -34,6 +36,8 @@ struct EnumTraits<ui::mojom::ClipboardType, ui::ClipboardType> {
       case ui::mojom::ClipboardType::DRAG:
         *out = ui::CLIPBOARD_TYPE_DRAG;
         return true;
+      default:
+        __builtin_abort ();
     }
     NOTREACHED();
     return false;
diff --git chromium-71.0.3578.30/ui/gfx/skia_vector_animation.cc chromium-71.0.3578.30/ui/gfx/skia_vector_animation.cc
index 8e16dc9dcc..06e35f4f0c 100644
--- chromium-71.0.3578.30/ui/gfx/skia_vector_animation.cc
+++ chromium-71.0.3578.30/ui/gfx/skia_vector_animation.cc
@@ -162,6 +162,8 @@ float SkiaVectorAnimation::GetCurrentProgress() const {
       // yet
       DCHECK(timer_control_);
       return timer_control_->GetNormalizedCurrentCycleProgress();
+    default:
+      __builtin_abort ();
   }
 }
 
diff --git chromium-71.0.3578.30/ui/views/layout/layout_provider.cc chromium-71.0.3578.30/ui/views/layout/layout_provider.cc
index 310d695c17..8da6558654 100644
--- chromium-71.0.3578.30/ui/views/layout/layout_provider.cc
+++ chromium-71.0.3578.30/ui/views/layout/layout_provider.cc
@@ -156,6 +156,8 @@ int LayoutProvider::GetCornerRadiusMetric(EmphasisMetric emphasis_metric,
       return is_touch ? 8 : 4;
     case EMPHASIS_MAXIMUM:
       return is_touch ? std::min(size.width(), size.height()) / 2 : 4;
+    default:
+      __builtin_abort ();
   }
 }
 
