diff -up rustc-1.80.1-src/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp.omv~ rustc-1.80.1-src/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp
--- rustc-1.80.1-src/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp.omv~	2024-08-08 21:19:59.224556355 +0200
+++ rustc-1.80.1-src/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp	2024-08-08 21:20:20.164778828 +0200
@@ -1576,7 +1576,7 @@ extern "C" LLVMValueRef LLVMRustGetInstr
 }
 
 extern "C" LLVMValueRef LLVMRustGetInstrProfMCDCCondBitmapIntrinsic(LLVMModuleRef M) {
-#if LLVM_VERSION_GE(18, 0)
+#if LLVM_VERSION_GE(18, 0) && LLVM_VERSION_LT(19, 0)
   return wrap(llvm::Intrinsic::getDeclaration(
       unwrap(M), llvm::Intrinsic::instrprof_mcdc_condbitmap_update));
 #else
