--- a/Makefile.in	2016-03-04 04:47:47.000000000 +0700
+++ b/Makefile.in	2016-03-10 14:10:53.289123389 +0700
@@ -228,6 +228,9 @@
 SYMBOL_INDEX_NAME = \
   $(MOZ_APP_NAME)-$(MOZ_APP_VERSION)-$(OS_TARGET)-$(BUILDID)-$(CPU_ARCH)$(EXTRA_BUILDID)-symbols.txt
 
+pgo-profile-run:
+	$(PYTHON) $(topsrcdir)/build/pgo/profileserver.py $(EXTRA_TEST_ARGS)
+
 buildsymbols:
 ifdef MOZ_CRASHREPORTER
 	echo building symbol store
