--- firefox-46.0.1/extensions/auth/nsAuthSambaNTLM.cpp.rhbz-1173156	2016-05-03 06:31:11.000000000 +0100
+++ firefox-46.0.1/extensions/auth/nsAuthSambaNTLM.cpp	2016-05-04 10:18:53.157429123 +0100
@@ -256,7 +256,8 @@ nsAuthSambaNTLM::GetNextToken(const void
     nsCString line;
     if (!ReadLine(mFromChildFD, line))
         return NS_ERROR_FAILURE;
-    if (!StringBeginsWith(line, NS_LITERAL_CSTRING("KK "))) {
+    if (!StringBeginsWith(line, NS_LITERAL_CSTRING("KK ")) &&
+        !StringBeginsWith(line, NS_LITERAL_CSTRING("AF "))) {
         // Something went wrong. Perhaps no credentials are accessible.
         return NS_ERROR_FAILURE;
     }
