diff -up firefox-73.0/media/libaom/moz.build.old firefox-73.0/media/libaom/moz.build
--- firefox-73.0/media/libaom/moz.build.old	2020-02-07 23:13:28.000000000 +0200
+++ firefox-73.0/media/libaom/moz.build	2020-02-17 10:30:08.509805092 +0200
@@ -55,7 +55,7 @@ elif CONFIG['CPU_ARCH'] == 'arm':
 
     for f in SOURCES:
         if f.endswith('neon.c'):
-            SOURCES[f].flags += CONFIG['VPX_ASFLAGS']
+            SOURCES[f].flags += CONFIG['NEON_FLAGS']
 
     if CONFIG['OS_TARGET'] == 'Android':
         # For cpu-features.h
